---
title: "Repair_Seq_Analysis"
output: html_notebook
---

```{r}
#fig. S1. Genomics of genome repair hotspots in post-mitotic human neurons.
#(F) Further examples of DNA repair hotspots in the DLG2 and the XRCC5 locus.


```

```{r}
library("DiffBind")
library("ChIPpeakAnno")
library("ChIPseeker")
library("pqsfinder")
library("motifRG")
library("BSgenome.Hsapiens.UCSC.hg38")
library("EnsDb.Hsapiens.v86")
library("TxDb.Hsapiens.UCSC.hg38.knownGene")
library("clusterProfiler")
library("ReactomePA")
library("org.Hs.eg.db")
library("viridis")
library("ggplot2")
library("ggExtra")
library("ggpmisc")



txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
anno86 <- annoGR(EnsDb.Hsapiens.v86)

```

```{r}
### Load new dba and bcb objects.
load("repairseq_dbaV2.rdata")
load("repairseq_bcbV2.rda")

# New masks basic 1D Peaks Common across H1 and H9
dba_H1H9_mask <- dba.mask(DBAobject,DBA_ID, c("H1","H9"))
dba_EdU_mask <- dba.mask(DBAobject,DBA_CONDITION, "EdU", mask=dba_H1H9_mask, merge="and")
dba_EdU_1Day_mask <- dba.mask(DBAobject,DBA_FACTOR,"1Day",mask=dba_EdU_mask, merge="and")
dba_EdU_1Day_None_mask <- dba.mask(DBAobject,DBA_TREATMENT,"None",mask=dba_EdU_1Day_mask, merge="and")
dba_EdU_1Day_None_ESCin_mask <- dba.mask(DBAobject,DBA_TISSUE,"ESC-iN",mask=dba_EdU_1Day_None_mask, merge="and")
dba_1Day_EdU_None_ESCin <- dba(DBAobject,dba_EdU_1Day_None_ESCin_mask)
dba_1D <- dba.contrast(dba_1Day_EdU_None_ESCin, categories=DBA_ID, minMembers = 2)
dba_1D <- dba.analyze(dba_1D, method=DBA_DESEQ2)
dba_1D_Report <- dba.report(dba_1D, contrast=1, th=1, bUsePval=TRUE)
dba_1D_Common <- dba_1D_Report[dba_1D_Report$`p-value` > .05]

#Annotate 1D peaks and plot genomic distribition
peakAnno_1D_Common <- annotatePeak(dba_1D_Common, tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db")
```

```{r}
#(B) Reproducibility between biological replicates of H1 and H9 ESC-iNs.

df_1d_H1_vs_H9 <- as.data.frame(dba_1D_Report)
##Convert Diffbind object to GRanges object
gr <- makeGRangesFromDataFrame(df_1d_H1_vs_H9,keep.extra.columns = TRUE)  
gr <- keepStandardChromosomes(gr, pruning.mode="coarse") ##Remove dumb contigs
gr <- dropSeqlevels(gr,  c("chrY"), pruning.mode="coarse") ##Remove Sex Chomosomes

H9 <- dba.peakset(dba_1Day_EdU_None_ESCin, 1:2, minOverlap=1, bRetrieve=TRUE)
H1 <- dba.peakset(dba_1Day_EdU_None_ESCin, 3:4, minOverlap=1, bRetrieve=TRUE)

df <- as.data.frame(H9)
df2 <- as.data.frame(H1)
df3 <- as.data.frame(gr)
my.formula <- y ~ x
pdf("H9_Replicate_Correlation.pdf")
ggplot(df, aes(x=H9, y=H9.1)) + geom_point(size=.2) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + stat_poly_eq(formula = my.formula, aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), parse = TRUE) + xlab("H9 ESC-iN Replicate 1") + ylab("H9 ESC-iN Replicate 2") + xlim(0,500) + ylim(0,500)
dev.off()

pdf("H1_Replicate_Correlation.pdf")
my.formula <- y ~ x
ggplot(df2, aes(x=H1, y=H1.1)) + geom_point(size=.2) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + stat_poly_eq(formula = my.formula, aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), parse = TRUE) + xlab("H1 ESC-iN Replicate 1") + ylab("H1 ESC-iN Replicate 2") + xlim(0,500) + ylim(0,500)
dev.off()

pdf("H1_H9_Correlation.pdf")
my.formula <- y ~ x
ggplot(df, aes(x=H1_Average, y=H9_Average)) + geom_point(size=.2) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + stat_poly_eq(formula = my.formula, aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), parse = TRUE) + xlab("H1") + ylab("H9") + xlim(0,500) + ylim(0,500)
dev.off()
```



```{r}

#(E) Genome annotations for DNA repair hotspots.
pdf("H1H9_1D_Common_Peaks_GenomicDistribution.pdf")
plotAnnoPie(peakAnno_1D_Common) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
dev.off()

#Look at functional enrichemnt of genes with 1D peaks
gene <- seq2gene(dba_1D_Common, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb, sameStrand=FALSE)
ego <- enrichGO(gene=gene, OrgDb=org.Hs.eg.db, ont="BP", pvalueCutoff =0.05)
pdf("H1H9_1D_Common_Peaks_GeneOntolgy.pdf")
barplot(ego, showCategory=20) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
dev.off()
cnetplot(ego, categorySize="pvalue", foldChange=gene)

```

```{r}
#Genome map of DNA repair hotspots in ESC-iNs.
library("ggbio")
library("smoother")
df_1d_H1_vs_H9 <- as.data.frame(dba_1D_Common)
gr <- makeGRangesFromDataFrame(df_1d_H1_vs_H9,keep.extra.columns = TRUE)  
gr <- keepStandardChromosomes(gr, pruning.mode="coarse") ##Remove dumb contigs
gr <- dropSeqlevels(gr,  c("chrY"), pruning.mode="coarse") ##Remove Sex Chomosomes
b <- c(248956422,242193529,198295559,190214555,181538259,170805979,159345973,145138636,138394717,133797422,135086622,133275309,114364328,107043718,101991189,90338345,83257441,80373285,58617616,64444167,46709983,50818468,156040895)
seqlengths(gr) <- b
score <- GenomicRanges::coverage(gr, weight="Conc")
bins <- GenomicRanges::tileGenome(GenomeInfoDb::seqlengths(gr), tilewidth=100000,cut.last.tile.in.chrom=TRUE)
binned_data <- GenomicRanges::binnedAverage(bins, score, "average_score")
binned_data$smooth_avg <- smth(binned_data$average_score,window = 10,method = "gaussian")

pdf("Chromosomes_Plot_100KB.pdf", paper="USr")
autoplot(binned_data, layout = "karyogram", aes(color = smooth_avg, fill = smooth_avg), na.value="grey0", chr="chrX")  +scale_colour_gradient2(low = "grey0", mid = "white",
high = "#662d91", midpoint = .05, space = "Lab", na.value = "grey0", guide = "colourbar") +scale_fill_gradient2(low = "grey0", mid = "white",
high = "#662d91", midpoint = .05, space = "Lab", na.value = "grey0", guide = "colourbar")
dev.off()

ggplot(binned_data, aes(x=c(1:binned_data@seqinfo@seqlengths[1]))) + geom_line(aes(y=binned_data$smooth_avg[1:248956422]))

length(binned_data$smooth_avg[1:248956422])
length(c(1:binned_data@seqinfo@seqlengths[1]))


```
```{r}
averagePerBin <- function(x, binsize, mcolnames=NULL)
{
     if (!is(x, "GenomicRanges"))
         stop("'x' must be a GenomicRanges object")
     if (any(is.na(seqlengths(x))))
         stop("'seqlengths(x)' contains NAs")
     bins <- IRangesList(lapply(seqlengths(x),
                                function(seqlen)
                                  IRanges(breakInChunks(seqlen, binsize))))
     ans <- as(bins, "GRanges")
     seqinfo(ans) <- seqinfo(x)
     if (is.null(mcolnames))
         return(ans)
     averageMCol <- function(colname)
     {
         cvg <- coverage(x, weight=colname)
         views_list <- RleViewsList(
                           lapply(names(cvg),
                               function(seqname)
                                   Views(cvg[[seqname]], bins[[seqname]])))
         unlist(viewMeans(views_list), use.names=FALSE)
     }
     mcols(ans) <- DataFrame(lapply(mcols(x)[mcolnames], averageMCol))
     ans
}
```

```{r}
#(D) More detailed view of DNA repair hotspots on Chromosome X. 

y = binned_chrX$smooth_avg
x = binned_chrX@ranges@start
my.max <- function(x) ifelse( !all(is.na(x)), max(x, na.rm=T), NA)
kp <- plotKaryotype(chromosomes="chrX", plot.type=2, genome="hg38")
kpPlotDensity(kp, all.genes, window.size = 0.5e6, data.panel = 2, col=muted("green"))
kpLines(kp, chr="chrX", x=x, y=y, col="#662d91", data.panel = 1)
kpAxis(kp, data.panel=1, ymax = my.max(y), r1=.3)
kpAxis(kp, data.panel=2)
