---
title: "MetaPsych: QC, Differential Expression, and Ontology"
author: "`r getOption('Patrick J Reed')`"
date: '`r Sys.Date()`'
fig_width: 24
output:
  html_document: 
    fig_width: 24
    keep_md: yes
    toc: yes
  pdf_document:
    fig_crop: no
params:
  alpha: 0.05
  organism: "Hs"
  gspecies: "hsa"
  species: "human"
  go_class: "BP"
  bcb_file: "bcb_repair-seq_H1H9.rda"
  data_dir: !r file.path("data", Sys.Date())
  results_dir: !r file.path("results", Sys.Date())
  design: formula("~ batch")
  dropbox_dir: null
  lfc_threshold: 0
bibliography: bibliography.bib
---

```{r setup, message=FALSE}
# Last modified 2018-05-08
bcbioRNASeq::prepareRNASeqTemplate()
source("_setup.R")

# Directory paths ==============================================================
invisible(mapply(
    FUN = dir.create,
    path = c(params$data_dir, params$results_dir),
    MoreArgs = list(showWarnings = FALSE, recursive = TRUE)
))

# Additional required packages =================================================
org_db <- paste("org", params$organism, "eg", "db", sep = ".")
packages <- c(
    org_db,
    "clusterProfiler",
    "DOSE",
    "pathview"
)
notInstalled <- setdiff(basename(packages), rownames(installed.packages()))
if (length(notInstalled)) {
    source("https://bioconductor.org/biocLite.R")
    biocLite(pkgs = notInstalled)
}
invisible(lapply(
    X = basename(packages),
    FUN = library,
    character.only = TRUE
))

# Prepare template =============================================================
bcbioRNASeq::prepareRNASeqTemplate()
source("_setup.R")

# Directory paths ==============================================================
kegg_plots_dir <- file.path(params$results_dir, "kegg_plots")
invisible(mapply(
    FUN = dir.create,
    path = c(params$data_dir, params$results_dir, kegg_plots_dir),
    MoreArgs = list(showWarnings = FALSE, recursive = TRUE)
))

# Load object ==================================================================
bcb_name <- load(params$bcb_file)
bcb <- get(bcb_name, inherits = FALSE)
stopifnot(is(bcb, "bcbioRNASeq"))
invisible(validObject(bcb))
```

```{r header, child="_header.Rmd"}
```



```{r sample_data}
sampleData(bcb)
```

[bcbio][] run data was imported from **`r metadata(bcb)$uploadDir`**.



# Count matrices

```{r counts}
raw_counts <- counts(bcb, normalized = FALSE)
# DESeq2 normalized counts
normalized_counts <- counts(bcb, normalized = TRUE)
# Transcripts per million
tpm <- counts(bcb, normalized = "tpm")
saveData(raw_counts, normalized_counts, tpm, dir = params$data_dir)
writeCounts(raw_counts, normalized_counts, tpm, dir = params$results_dir)
```

The results are saved as gzip-compressed comma separated values (CSV). Gzip compression is natively supported on [macOS][] and Linux-based operating systems. If you're running Windows, we recommend installing [7-Zip][]. CSV files can be opened in [Excel][] or [RStudio][].

- [`normalized_counts.csv.gz`](`r file.path(params$results_dir, "normalized_counts.csv.gz")`): Use to evaluate individual genes and/or generate plots. These counts are normalized for the variation in sequencing depth across samples.
- [`tpm.csv.gz`](`r file.path(params$results_dir, "tpm.csv.gz")`): Transcripts per million, scaled by length and also suitable for plotting.
- [`raw_counts.csv.gz`](`r file.path(params$results_dir, "raw_counts.csv.gz")`): Only use to perform a new differential expression analysis. These counts will vary across samples due to differences in sequencing depth, and have not been normalized. Do not use this file for plotting genes.



# Read metrics {.tabset}

## Total reads

```{r}
bcb$celltypeBroad <- bcb$cohort
```

```{r, plot_total_reads, fig.width=24}
plotTotalReads(bcb, flip = FALSE)  + theme(axis.title.x=element_blank(),axis.text.x=element_blank(), axis.ticks.x=element_blank())
```


## Mapped reads

The number of mapped reads should correspond to the number of total reads.

```{r plot_mapped_reads, fig.width=24}
plotMappedReads(bcb, flip=FALSE)  + theme(axis.title.x=element_blank(),axis.text.x=element_blank(), axis.ticks.x=element_blank())
```


## Mapping rate

The genomic mapping rate represents the percentage of reads mapping to the reference genome. Low mapping rates are indicative of sample contamination, poor sequencing quality or other artifacts.

```{r plot_mapping_rate, fig.width=24}
plotMappingRate(bcb, flip=FALSE, limit = 95) + theme(axis.title.x=element_blank(),axis.text.x=element_blank(), axis.ticks.x=element_blank())
```


## Number of genes detected

```{r plot_genes_detected, fig.width=24}
plotGenesDetected(bcb, flip=FALSE, limit = 25000) + theme(axis.title.x=element_blank(),axis.text.x=element_blank(), axis.ticks.x=element_blank())
```

## Gene detection saturation

We should observe a linear trend in the number of genes detected with the number of mapped reads, which indicates that the sample input was not overloaded.

```{r plot_gene_saturation}
plotGeneSaturation(bcb, interestingGroups=c("projectLead"))
```



## Counts per gene

Generally, we expect similar count spreads for all genes between samples unless the library sizes or total RNA expression are different. The log10 TMM-normalized counts per gene normalization method [@Robinson:2010dd] equates the overall expression levels of genes between samples under the assumption that the majority of them are not differentially expressed. Therefore, by normalizing for total RNA expression by sample, we expect the spread of the log10 TMM-normalized counts per gene to be similar for every sample.

```{r plot_counts_per_gene}
plotCountsPerGene(bcb, flip=FALSE, normalized="rlog")
```


```{r plot_biotype, eval="geneBiotype" %in% colnames(rowData(bcb)), echo=FALSE}
mdHeader("TPM per biotype", level = 2)
asis_output("Different RNA-seq processing methods can preferentially capture a subset of the RNA species from the total RNA. For example, polyA selection should select for mostly coding genes and skip a large percentage of non-polyA non-coding RNA. Here we make boxplots of the TPM for the top 12 biotypes with the most genes assigned to them for each sample.")

keep_biotypes <- rowData(bcb) %>%
    as.data.frame() %>%
    group_by(geneBiotype) %>%
    summarise(nBiotype = n()) %>%
    arrange(-nBiotype) %>%
    top_n(12, wt = nBiotype) %>%
    pull(geneBiotype) %>%
    droplevels()
biotype_tpm <- tpm(bcb) %>%
    as.data.frame() %>%
    rownames_to_column("geneID") %>%
    gather(key = sampleID, value = tpm, -geneID) %>%
    left_join(as.data.frame(rowData(bcb)), by = "geneID") %>%
    filter(geneBiotype %in% keep_biotypes) %>%
    filter(tpm > 0)

ggplot(
    data = biotype_tpm,
    mapping = aes(
        x = sampleID,
        y = tpm,
        fill = sampleID
    )
) +
    geom_violin(
        color = "black",
        scale = "area"
    ) +
    scale_y_log10() +
    facet_wrap(~geneBiotype, scales = "free_y") +
    labs(
        title = "tpm per biotype",
        x = NULL,
        y = "transcripts per million (tpm)"
    ) +
    guides(fill = FALSE) +
    theme(axis.text.x = element_text(angle = 90L, hjust = 1L, vjust = 0.5))
```


```{r plot_broad_class, eval="broadClass" %in% colnames(rowData(bcb)), echo=FALSE}
mdHeader("TPM per broad biotype class", level = 2)
asis_output("The Ensembl biotype clasifications are too specific to plot them all. Here we have grouped the biotypes into broad classes and plot boxplots of the TPM for each sample.")

broad_tpm <- tpm(bcb) %>%
    as.data.frame() %>%
    rownames_to_column("geneID") %>%
    gather(key = sampleID, value = tpm, -geneID) %>%
    left_join(as.data.frame(rowData(bcb)), by = "geneID") %>%
    filter(!is.na(broadClass)) %>%
    filter(tpm > 0)

ggplot(
    data = broad_tpm,
    mapping = aes(
        x = sampleID,
        y = tpm,
        fill = sampleID
    )
) +
    geom_violin(
        color = "black",
        scale = "area"
    ) +
    facet_wrap(~broadClass, scales = "free_y") +
    scale_y_log10() +
    labs(
        title = "tpm per broad biotype class",
        x = NULL,
        y = "transcripts per million (tpm)"
    ) +
    guides(fill = FALSE) +
    theme(axis.text.x = element_text(angle = 90L, hjust = 1L, vjust = 0.5))
```


## Count density

Generally, we expect similar count spreads for all genes between samples unless the total expressed RNA per sample is different.

```{r plot_count_density}
plotCountDensity(bcb, normalized = "tpm")
```



# Fit modeling

Several quality metrics are first assessed to explore the fit of the model, before differential expression analysis is performed.


## Variance stabilization

The plots below show the standard deviation of normalized counts (`normalized_counts`) using `log2()`, `rlog()`, and variance stabilizing (`vst()`) transformations by `rank(mean)`. The transformations greatly reduce the standard deviation, with `rlog()` stabilizing the variance best across the mean.

```{r plot_mean_sd, fig.height=18, fig.width=6}
plotMeanSD(bcb, orientation = "vertical")
```


## Dispersion

The following plot shows the dispersion by mean of normalized counts. We expect the dispersion to decrease as the mean of normalized counts increases.

```{r plot_disp_ests}
plotDispEsts(bcb)
```



# Sample similarity analysis

Before performing similarity analysis, we transform counts to log2, which acts to minimize large differences in sequencing depth and helps normalize all samples to a similar dynamic range. For RNA-seq count data, variance increases with the mean. Logarithmic transformation of normalized count values with a small pseudocount will account for large variations seen between the highest expressing genes so that these genes won't dominate the PCA plots. However, due to the strong noise among low count values due to Poisson, the general log2 transformation will amplify this noise, and instead, low count genes will now dominate the PCA plots. So instead, we use a regularized log ("rlog"; base 2) transformation that gives similar results for high counts as a log2 transformation but also shrinks the values of low counts towards the genesâ€™ average across samples. We do this with the `rlog()` function in the [DESeq2][] package [@DESeq2], which we will later use for differential gene expression analysis.


## Principal component analysis (PCA) {.tabset}

PCA [@Jolliffe:2002wx] is a multivariate technique that allows us to summarize the systematic patterns of variations in the data. PCA takes the expression levels for genes and transforms it in principal component space, reducing each sample into one point. Thereby, we can separate samples by expression variation, and identify potential sample outliers. The PCA plot is a way to look at how samples are clustering.


```{r plot_pca}
markdownHeader("labeled", level = 3)
plotPCA(bcb, label = FALSE, interestingGroups=c("celltypeBroad"))
```

```{r plot_pca}
markdownHeader("labeled", level = 3)
plotPCA(bcb, label = FALSE, interestingGroups=c("sePESequencing"))
```
```{r plot_pca}
markdownHeader("labeled", level = 3)
plotPCA(bcb, label = FALSE, interestingGroups=c("celltypeSpecific"))
```

```{r plot_pca}
markdownHeader("labeled", level = 3)
plotPCA(bcb, label = FALSE, interestingGroups=c("cohort"))
```

```{r plot_pca}
markdownHeader("labeled", level = 3)
plotPCA(bcb, label = FALSE, interestingGroups=c("krakenClas"))
```

```{r plot_tsne}
markdownHeader("unlabeled", level = 3)
tsne_out <- Rtsne(t(bcb@assays$data$normalized), theta=0, perplexity=30, max_iter = 5000)
tsne_plot <- data.frame(x = tsne_out$Y[,1], y = tsne_out$Y[,2], col = bcb@colData$celltypeBroad)
ggplot(tsne_plot) + geom_point(aes(x=x, y=y, color=col))
```

```{r plot_tsne}
ggplot(tsne_plot) + geom_point(aes(x=x, y=y, color=bcb@colData$sePESequencing))
```
## Covariates correlation with PCs

When multiple factors may influence the results of a given experiment, it is useful to assess which of them is responsible for the most variance as determined by PCA. We adapted the method described by Daily et al. where they integrated a method to correlate covariates with principal components values to determine the importance of each factor.

Here we are showing the correlational analysis of the rlog transformed count data's principal components with the metadata covariates of interest. Significant correlations (FDR < 0.1) are shaded from blue (anti-correlated) to orange (correlated), with non-significant correlations shaded in gray.

```{r plot_pca_covariates}
plotPCACovariates(bcb)
```


## Hierarchical clustering

Inter-correlation analysis (ICA) is another way to look at how well samples cluster by plotting the correlation between the expression profiles of the samples.

```{r plot_correlation_heatmap}
plotCorrelationHeatmap(bcb, method = "pearson", interestingGroups=c("celltypeSpecific","cohort","sePESequencing"), showRownames = FALSE, showColnames = FALSE)
```

```{r dds, results="hide"}
dds <- as(bcb, "DESeqDataSet")
dds$group <- factor(paste0(dds$cohort, dds$celltypeBroad, dds$phenotype))
design(dds) <- ~ group
dds <- DESeq(dds, parallel = TRUE) 
resultsNames(dds)
# Alternatively, can use `rlog()`
rld <- varianceStabilizingTransformation(dds)
```

```{r alpha_summary, results="asis"}
alphaSummary(dds)
```

```{r results}
res <- results(dds, contrast=c("group","iPSC.iNAD","iPSC.iNCTL"))
```

We performed the analysis using a BH adjusted *P* value cutoff of `r params$alpha` and a log fold-change (LFC) ratio cutoff of `r params$lfc_threshold`.



# Plots

## Mean average (MA)

An MA plot compares transformed counts on `M` (log ratio) and `A` (mean average) scales [@Yang:2002ty].

```{r plot_ma2}
gene2symbol <- gene2symbol(bcb)
plotMeanAverage(res, gene2symbol = gene2symbol)
# Alternate plot
#DESeq2::plotMA(res)
```

## Volcano

A volcano plot compares significance (BH-adjusted *P* value) against fold change (log2) [@Cui:2003kh; @Li:2014fv]. Genes in the green box with text labels have an adjusted *P* value are likely to be the top candidate genes of interest.

```{r plot_volcano2}
plotVolcano(res)
```

## Heatmap

This plot shows only differentially expressed genes on a per-sample basis. We have scaled the data by row and used the `ward.D2` method for clustering [@WardJr:1963eu].
```{r plot_deg_heatmap2}
# help("pheatmap", "pheatmap")
plotDEGHeatmap(
    results = res,
    counts = rld,
    clusteringMethod = "ward.D2",
    scale = "row",
    clusterRows = TRUE,
    showColnames = TRUE,
    interestingGroups = c("phenotype")#,
    #lfcThreshold = .5
)
```

```{r plot_deg_pca2}
plotDEGPCA(
    label = FALSE,
    results = res,
    counts = rld,
    interestingGroups = c("phenotype")
)
```

# Results tables

```{r results_tables2, results="asis"}
res_tbl <- resultsTables(
    results = res,
    counts = dds,
    lfcThreshold = 1,
    write = TRUE,
    summary = TRUE,
    headerLevel = 2,
    dir = params$results_dir,
    dropboxDir = params$dropbox_dir
)
saveData(res, res_tbl, dir = params$data_dir)
```

Differentially expressed gene (DEG) tables are sorted by BH-adjusted P value, and contain the following columns:

- `baseMean`: Mean of the normalized counts per gene for all samples.
- `log2FoldChange`: log2 fold change.
- `lfcSE`: log2 standard error.
- `stat`: Wald statistic.
- `pvalue`: Walt test *P* value.
- `padj`: BH adjusted Wald test *P* value (corrected for multiple comparisons; aka FDR).


## Top tables

Only the top up- and down-regulated genes (arranged by log2 fold change) are shown.

```{r top_tables2, results="asis"}
topTables(res_tbl)
```

```{r results}
# Subset NA adjusted P values
#res <- res[!is.na(res$padj), , drop = FALSE]
alpha <- metadata(res)$alpha

all_genes <- rownames(res)
# help("significants", "DEGreport")
sig_genes <- significants(res, fc = params$lfc_threshold, padj = alpha)

# All genes containing a P value, ordered by FDR
res_df <- res %>%
    as("DataFrame") %>%
    .[order(.$padj), c("log2FoldChange", "padj")]

# Significant genes 
sig_res_df <- res_df %>%
    .[which(rownames(.) %in% sig_genes), , drop = FALSE]

lfc_vec <- sig_res_df$log2FoldChange
names(lfc_vec) <- rownames(sig_res_df)
# Sort from upregulated to downregulated
lfc_vec <- sort(lfc_vec, decreasing = TRUE)
```



# GO enrichment analysis

[Gene Ontology (GO)][GO] term enrichment is a technique for interpreting sets of genes making use of the [Gene Ontology][GO] system of classification, in which genes are assigned to a set of predefined bins depending on their functional characteristics.

```{r enrich_go}
# help("enrichGO", "clusterProfiler")
enrich_go <- enrichGO(
    gene = sig_genes,
    OrgDb = org_db,
    keyType = "ENSEMBL",
    ont = params$go_class,
    universe = all_genes,
    qvalueCutoff = 0.05,
    readable = TRUE
)
enrich_go_result <- enrich_go %>%
    slot("result") %>%
    as_tibble() %>%
    camel()
saveData(enrich_go, enrich_go_result, dir = params$data_dir)

write_csv(
    enrich_go_result,
    path = file.path(
        params$results_dir,
        paste0(
            paste(
                "go",
                tolower(params$go_class),
                "clusterprofiler",
                "padj",
                alpha,
                "lfc",
                params$lfc_threshold,
                sep = "_"
            ),
            ".csv.gz"
        )
    )
)

enrich_go_result
```


## Dot plot

```{r dot_plot, fig.width=8}
dotplot(enrich_go)
```


## GO terms map

```{r enrich_map, fig.width=8, fig.height=8}
# help("enrichMap", "DOSE")
emapplot(enrich_go, n = 25, vertex.label.cex = 0.5)
```


## Gene map

In order to consider the potentially biological complexities in which a gene may belong to multiple annotation categories, and provide information of numeric changes if available.

Here we are plotting genes colored by LFC for top 5 most significant [GO][] terms.

```{r cnetplot, fig.width=8, fig.height=8}
# help("cnetplot", "clusterProfiler")
cnetplot(
    enrich_go,
    foldChange = lfc_vec,
    vertex.label.cex = 0.5
)
```


## GO GSEA analysis

A common approach in analyzing gene expression profiles was identifying differential expressed genes that are deemed interesting. The enrichment analysis we demonstrated previously were based on these differentially expressed genes. This approach will find genes where the difference is large, but it will not detect a situation where the difference is small, but evidenced in coordinated way in a set of related genes. [Gene Set Enrichment Analysis (GSEA)][GSEA] directly addresses this limitation. All genes can be used in [GSEA][]; [GSEA][] aggregates the per gene statistics across genes within a gene set, therefore making it possible to detect situations where all genes in a predefined set change in a small but coordinated way. Since it is likely that many relevant phenotypic differences are manifested by small but consistent changes in a set of genes.

```{r gsea_go}
# Prepare the gene list. Here we're subtracting the adjusted P value from 1.
gene_list <- 1 - res_df$padj
names(gene_list) <- row.names(res_df)

# Now run GSEA
# help("gseGO", "clusterProfiler")
gsea_go <- gseGO(
    geneList = gene_list,
    ont = params$go_class,
    OrgDb = org_db,
    keyType = "ENSEMBL")
gsea_go_resultL <- gsea_go %>%
    slot("result") %>%
    as_tibble() %>%
    camel()
saveData(gsea_go, gsea_go_result, dir = params$data_dir)

write_csv(
    gsea_go,
    path = file.path(
        params$results_dir,
        paste0(
            paste(
                "gsea",
                "clusterprofiler",
                "padj",
                alpha,
                "lfc",
                params$lfc_threshold,
                sep = "_"
            ),
            ".csv.gz"
        )
    )
)

gsea_go_result
```

# KEGG enrichment analysis

## Map gene IDs to Entrez IDs

[Entrez][] IDs are required for [Kyoto Encyclopedia of Genes and Genomes (KEGG)][KEGG] analysis. Here we are defining 1:1 mappings of the [Ensembl][] gene IDs to [Entrez][] IDs. For genes that map to multiple [Entrez][] IDs, we are using the oldest [Entrez][] ID to define the 1:1 mapping.

```{r entrez}
# Entrez IDs here are obtained from AnnotationHub using ensembldb
entrez <- rowData(bcb)[all_genes, "entrezID", drop = TRUE]
stopifnot(is.list(entrez))
stopifnot(!is.null(names(entrez)))
# For genes that don't map 1:1 with Entrez, use the oldest Entrez ID
entrez <- lapply(entrez, function(x) {
    if (all(is.na(x))) {
        NULL
    } else {
        sort(x)[[1L]]
    }
})
entrez <- Filter(Negate(is.null), entrez)

# Ensembl gene IDs are names; Entrez gene ID are values
gene2entrez <- unlist(entrez)
stopifnot(is.integer(gene2entrez))

entrez_res <- res[names(gene2entrez), ]
rownames(entrez_res) <- as.character(gene2entrez)

all_entrez <- rownames(entrez_res)
# help("significants", "DEGreport")
sig_entrez <- significants(entrez_res, fc = params$lfc_threshold, padj = alpha)

# All genes containing a P value, ordered by FDR
entrez_res_df <- entrez_res %>%
    as("DataFrame") %>%
    .[order(.$padj), c("log2FoldChange", "padj")]

# Significant genes 
sig_entrez_res_df <- entrez_res_df %>%
    .[which(rownames(.) %in% sig_entrez), ]

# Extract the fold changes
entrez_lfc_vec <- sig_entrez_res_df$log2FoldChange
names(entrez_lfc_vec) <- rownames(sig_entrez_res_df)
# Sort from upregulated to downregulated
entrez_lfc_vec <- sort(entrez_lfc_vec, decreasing = TRUE)

saveData(gene2entrez, entrez_res, dir = params$data_dir)
```



```{r footer, child="_footer.Rmd"}
```
