---
title: "Repair_Seq_Analysis"
output: html_notebook
---

```{r}
#Fig. S2. CRISPR/Cas9 generates novel DNA repair hotspots in post-mitotic human neurons.
#(A) CRISPR/Cas9 induced DNA repair hotspots in C9orf72 and RANBP17.
#(B) RPMs (C9orf72, ERG, FOXO1, GATA3, HOMER1, RANBP17, SIRT6) Cas9 vs Cas9+guides Box plot (2xH1 + 2xH9s)


```

```{r}
library("DiffBind")
library("ChIPpeakAnno")
library("ChIPseeker")
library("pqsfinder")
library("motifRG")
library("BSgenome.Hsapiens.UCSC.hg38")
library("EnsDb.Hsapiens.v86")
library("TxDb.Hsapiens.UCSC.hg38.knownGene")
library("clusterProfiler")
library("ReactomePA")
library("org.Hs.eg.db")
library("viridis")
library("ggplot2")
library("ggExtra")
library("ggpmisc")



txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
anno86 <- annoGR(EnsDb.Hsapiens.v86)

```

```{r}
### Load new dba and bcb objects.
load("repairseq_dbaV2.rdata")
load("repairseq_bcbV2.rda")

# New masks basic 1D Peaks Common across H1 and H9
dba_H1H9_mask <- dba.mask(DBAobject,DBA_ID, c("H1","H9"))
dba_EdU_mask <- dba.mask(DBAobject,DBA_CONDITION, "EdU", mask=dba_H1H9_mask, merge="and")
dba_EdU_1Day_mask <- dba.mask(DBAobject,DBA_FACTOR,"1Day",mask=dba_EdU_mask, merge="and")
dba_EdU_1Day_None_mask <- dba.mask(DBAobject,DBA_TREATMENT,"None",mask=dba_EdU_1Day_mask, merge="and")
dba_EdU_1Day_None_ESCin_mask <- dba.mask(DBAobject,DBA_TISSUE,"ESC-iN",mask=dba_EdU_1Day_None_mask, merge="and")
dba_1Day_EdU_None_ESCin <- dba(DBAobject,dba_EdU_1Day_None_ESCin_mask)
dba_1D <- dba.contrast(dba_1Day_EdU_None_ESCin, categories=DBA_ID, minMembers = 2)
dba_1D <- dba.analyze(dba_1D, method=DBA_DESEQ2)
dba_1D_Report <- dba.report(dba_1D, contrast=1, th=1, bUsePval=TRUE)
dba_1D_Common <- dba_1D_Report[dba_1D_Report$`p-value` > .05]

#Annotate 1D peaks and plot genomic distribition
peakAnno_1D_Common <- annotatePeak(dba_1D_Common, tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db")
plotAnnoPie(peakAnno_1D_Common) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))

#Look at functional enrichemnt of genes with 1D peaks
gene <- seq2gene(dba_1D_Common, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb, sameStrand=FALSE)
ego <- enrichGO(gene=gene, OrgDb=org.Hs.eg.db, ont="BP", pvalueCutoff =0.05)
dotplot(ego) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
cnetplot(ego, categorySize="pvalue", foldChange=gene)

##H1 vs H9 correlation


df_1d_H1_vs_H9 <- as.data.frame(dba_1D_Report)
##Convert Diffbind object to GRanges object
gr <- makeGRangesFromDataFrame(df_1d_H1_vs_H9,keep.extra.columns = TRUE)  
gr <- keepStandardChromosomes(gr, pruning.mode="coarse") ##Remove dumb contigs
gr <- dropSeqlevels(gr,  c("chrX","chrY"), pruning.mode="coarse") ##Remove Sex Chomosomes
df <- as.data.frame(gr)
my.formula <- y ~ x
ggplot(df, aes(x=Conc_H1, y=Conc_H9)) + geom_point(size=.2) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + stat_poly_eq(formula = my.formula, aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), parse = TRUE) + xlim(2,10)  + ylim(2,10)


```


