---
title: "RepairedSeq CRISPR"
author: "`r getOption('author')`"
date: "`r Sys.Date()`"
bibliography: bibliography.bib
# Using bcbioRNASeq::bcb_small dataset for example params here
params:
  # bcbioRNASeq object
  bcb_file: "repairseq_bcbV2.rda"
  # Design formula must contain columns defined in `colData()`
  design: !r formula("~ batch + treatment")
  # Multiple contrasts are supported
  alpha: 0.01
  lfc_threshold: 0
  data_dir: !r file.path(".", Sys.Date())
  results_dir: !r file.path(".", Sys.Date(), "differential_expression")
  dropbox_dir: NULL
---


```{r}
#Fig. S2. CRISPR/Cas9 generates novel DNA repair hotspots in post-mitotic human neurons.
#(A) CRISPR/Cas9 induced DNA repair hotspots in C9orf72 and RANBP17.
#(B) RPMs (C9orf72, ERG, FOXO1, GATA3, HOMER1, RANBP17, SIRT6) Cas9 vs Cas9+guides Box plot (2xH1 + 2xH9s)


```

```{r}
library("DiffBind")
library("ChIPpeakAnno")
library("ChIPseeker")
library("pqsfinder")
library("motifRG")
library("BSgenome.Hsapiens.UCSC.hg38")
library("EnsDb.Hsapiens.v86")
library("TxDb.Hsapiens.UCSC.hg38.knownGene")
library("clusterProfiler")
library("ReactomePA")
library("org.Hs.eg.db")
library("viridis")
library("ggplot2")
library("ggExtra")
library("ggpmisc")



txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
anno86 <- annoGR(EnsDb.Hsapiens.v86)

```

```{r}
### Load new dba and bcb objects.
load("repairseq_dbaV2.rdata")
load("repairseq_bcbV2.rda")

#(B) RPMs (C9orf72, ERG, FOXO1, GATA3, HOMER1, RANBP17, SIRT6) Cas9 vs Cas9+guides Box plot (2xH1 + 2xH9s)
dba_H1H9_mask <- dba.mask(DBAobject,DBA_ID, c("H1","H9"))
dba_EdU_mask <- dba.mask(DBAobject,DBA_CONDITION, "EdU", mask=dba_H1H9_mask, merge="and")
dba_EdU_CRISPR <- dba.mask(DBAobject,DBA_TREATMENT,c("Cas9_empty_14","Cas9_empty_21","Cas9_guides_14","Cas9_guides_21"),mask=dba_EdU_1Day_mask, merge="and")
dba_1Day_EdU_CRISPR <- dba(DBAobject,dba_EdU_CRISPR)
dba_1D_CRISPR_14 <- dba.contrast(dba_1Day_EdU_CRISPR, group1=dba_1Day_EdU_CRISPR$masks$Cas9_guides_14, group2 = dba_1Day_EdU_CRISPR$masks$Cas9_empty_14, minMembers = 2)

dba_1D_CRISPR_21 <- dba.contrast(dba_1Day_EdU_CRISPR, group1=dba_1Day_EdU_CRISPR$masks$Cas9_guides_21, group2 = dba_1Day_EdU_CRISPR$masks$Cas9_empty_21, minMembers = 2)

dba_1D_CRISPR_14 <- dba.analyze(dba_1D_CRISPR_14, method=DBA_DESEQ2)
dba_1D_CRISPR_14_Report <- dba.report(dba_1D_CRISPR_14, contrast=1, th=1, bUsePval=TRUE)
dba_1D_CRISPR_14_DE <- dba_1D_CRISPR_14_Report[dba_1D_CRISPR_14_Report$`p-value` < .05]

dba_1D_CRISPR_21 <- dba.analyze(dba_1D_CRISPR_21, method=DBA_DESEQ2)
dba_1D_CRISPR_21_Report <- dba.report(dba_1D_CRISPR_21, contrast=1, th=1, bUsePval=TRUE)
dba_1D_CRISPR_21_DE <- dba_1D_CRISPR_21_Report[dba_1D_CRISPR_21_Report$`p-value` < .05]

#Annotate 1D peaks and plot genomic distribition
peakAnno_1D_CRISPR_14_DE <- annotatePeak(dba_1D_CRISPR_14_DE, tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db", level = "gene")

peakAnno_1D_CRISPR_21_DE <- annotatePeak(dba_1D_CRISPR_21_DE, tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db", level = "gene")

peakAnno_1D_CRISPR_14_DE@anno$FDR[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "C9orf72"]
peakAnno_1D_CRISPR_14_DE@anno$FDR[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "ERG"]
peakAnno_1D_CRISPR_14_DE@anno$FDR[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "FOXO1"]
peakAnno_1D_CRISPR_14_DE@anno$FDR[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "GATA3"]
peakAnno_1D_CRISPR_14_DE@anno$FDR[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "HOMER1"]
peakAnno_1D_CRISPR_14_DE@anno$FDR[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "RANBP17"]
peakAnno_1D_CRISPR_14_DE@anno$FDR[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "SIRT6"]
peakAnno_1D_CRISPR_14_DE@anno$FDR[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "ADAM10"]


peakAnno_1D_CRISPR_21_DE@anno$FDR[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "C9orf72"]
peakAnno_1D_CRISPR_21_DE@anno$FDR[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "ERG"]
peakAnno_1D_CRISPR_21_DE@anno$FDR[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "FOXO1"]
peakAnno_1D_CRISPR_21_DE@anno$FDR[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "GATA3"]
peakAnno_1D_CRISPR_21_DE@anno$FDR[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "HOMER1"]
peakAnno_1D_CRISPR_21_DE@anno$FDR[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "RANBP17"]
peakAnno_1D_CRISPR_21_DE@anno$FDR[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "SIRT6"]
peakAnno_1D_CRISPR_21_DE@anno$FDR[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "ADAM10"]

peakAnno_1D_CRISPR_14_DE@anno$`p-value`[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "C9orf72"]
peakAnno_1D_CRISPR_14_DE@anno$`p-value`[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "ERG"]
peakAnno_1D_CRISPR_14_DE@anno$`p-value`[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "FOXO1"]
peakAnno_1D_CRISPR_14_DE@anno$`p-value`[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "GATA3"]
peakAnno_1D_CRISPR_14_DE@anno$`p-value`[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "HOMER1"]
peakAnno_1D_CRISPR_14_DE@anno$`p-value`[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "RANBP17"]
peakAnno_1D_CRISPR_14_DE@anno$`p-value`[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "SIRT6"]
peakAnno_1D_CRISPR_14_DE@anno$`p-value`[peakAnno_1D_CRISPR_14_DE@anno$SYMBOL == "ADAM10"]


peakAnno_1D_CRISPR_21_DE@anno$`p-value`[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "C9orf72"]
peakAnno_1D_CRISPR_21_DE@anno$`p-value`[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "ERG"]
peakAnno_1D_CRISPR_21_DE@anno$`p-value`[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "FOXO1"]
peakAnno_1D_CRISPR_21_DE@anno$`p-value`[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "GATA3"]
peakAnno_1D_CRISPR_21_DE@anno$`p-value`[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "HOMER1"]
peakAnno_1D_CRISPR_21_DE@anno$`p-value`[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "RANBP17"]
peakAnno_1D_CRISPR_21_DE@anno$`p-value`[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "SIRT6"]
peakAnno_1D_CRISPR_21_DE@anno$`p-value`[peakAnno_1D_CRISPR_21_DE@anno$SYMBOL == "ADAM10"]
```
```{r}
# H1 Crispr Analysis
dba_H1_mask <- dba.mask(DBAobject,DBA_ID, "H1")
dba_EdU_mask <- dba.mask(DBAobject,DBA_CONDITION, "EdU", mask=dba_H1_mask, merge="and")
dba_EdU_CRISPR <- dba.mask(DBAobject,DBA_TREATMENT,c("Cas9_empty_14","Cas9_empty_21","Cas9_guides_14","Cas9_guides_21"),mask=dba_EdU_mask, merge="and")
dba_1Day_EdU_CRISPR <- dba(DBAobject,dba_EdU_CRISPR)
dba_1D_CRISPR_14 <- dba.contrast(dba_1Day_EdU_CRISPR, group1=dba_1Day_EdU_CRISPR$masks$Cas9_guides_14, group2 = dba_1Day_EdU_CRISPR$masks$Cas9_empty_14, minMembers = 2)

dba_1D_CRISPR_21 <- dba.contrast(dba_1Day_EdU_CRISPR, group1=dba_1Day_EdU_CRISPR$masks$Cas9_guides_21, group2 = dba_1Day_EdU_CRISPR$masks$Cas9_empty_21, minMembers = 2)

dba_1D_CRISPR_14 <- dba.analyze(dba_1D_CRISPR_14, method=DBA_DESEQ2)
dba_1D_CRISPR_14_Report <- dba.report(dba_1D_CRISPR_14, contrast=1, th=1, bUsePval=TRUE)
dba_1D_CRISPR_14_DE <- dba_1D_CRISPR_14_Report[dba_1D_CRISPR_14_Report$`p-value` < .05]

dba_1D_CRISPR_21 <- dba.analyze(dba_1D_CRISPR_21, method=DBA_DESEQ2)
dba_1D_CRISPR_21_Report <- dba.report(dba_1D_CRISPR_21, contrast=1, th=1, bUsePval=TRUE)
dba_1D_CRISPR_21_DE <- dba_1D_CRISPR_21_Report[dba_1D_CRISPR_21_Report$`p-value` < .05]

peakAnno_H1_CRISPR_14_DE <- annotatePeak(dba_1D_CRISPR_14_DE, tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db", level = "gene")

df_H1_14 <- as.data.frame(peakAnno_H1_CRISPR_14_DE)
write.table(df_H1_14, file="peakAnno_H1_CRISPR_14_DiffBind.txt", sep="\t", col.names = TRUE, row.names = FALSE)


peakAnno_H1_CRISPR_21_DE <- annotatePeak(dba_1D_CRISPR_21_DE, tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db", level = "gene")

df_H1_21 <- as.data.frame(peakAnno_H1_CRISPR_21_DE)
write.table(df_H1_21, file="peakAnno_H1_CRISPR_21_DiffBind.txt", sep="\t", col.names = TRUE, row.names = FALSE)

# H9 Crispr Analysis
dba_H9_mask <- dba.mask(DBAobject,DBA_ID, "H9")
dba_EdU_mask <- dba.mask(DBAobject,DBA_CONDITION, "EdU", mask=dba_H9_mask, merge="and")
dba_EdU_CRISPR <- dba.mask(DBAobject,DBA_TREATMENT,c("Cas9_empty_14","Cas9_empty_21","Cas9_guides_14","Cas9_guides_21"),mask=dba_EdU_mask, merge="and")
dba_1Day_EdU_CRISPR <- dba(DBAobject,dba_EdU_CRISPR)
dba_1D_CRISPR_14 <- dba.contrast(dba_1Day_EdU_CRISPR, group1=dba_1Day_EdU_CRISPR$masks$Cas9_guides_14, group2 = dba_1Day_EdU_CRISPR$masks$Cas9_empty_14, minMembers = 2)

dba_1D_CRISPR_21 <- dba.contrast(dba_1Day_EdU_CRISPR, group1=dba_1Day_EdU_CRISPR$masks$Cas9_guides_21, group2 = dba_1Day_EdU_CRISPR$masks$Cas9_empty_21, minMembers = 2)

dba_1D_CRISPR_14 <- dba.analyze(dba_1D_CRISPR_14, method=DBA_DESEQ2)
dba_1D_CRISPR_14_Report <- dba.report(dba_1D_CRISPR_14, contrast=1, th=1, bUsePval=TRUE)
dba_1D_CRISPR_14_DE <- dba_1D_CRISPR_14_Report[dba_1D_CRISPR_14_Report$`p-value` < .05]

dba_1D_CRISPR_21 <- dba.analyze(dba_1D_CRISPR_21, method=DBA_DESEQ2)
dba_1D_CRISPR_21_Report <- dba.report(dba_1D_CRISPR_21, contrast=1, th=1, bUsePval=TRUE)
dba_1D_CRISPR_21_DE <- dba_1D_CRISPR_21_Report[dba_1D_CRISPR_21_Report$`p-value` < .05]

peakAnno_H9_CRISPR_14_DE <- annotatePeak(dba_1D_CRISPR_14_DE, tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db", level = "gene")

df_H9_14 <- as.data.frame(peakAnno_H9_CRISPR_14_DE)
write.table(df_H9_14, file="peakAnno_H9_CRISPR_14_DiffBind.txt", sep="\t", col.names = TRUE, row.names = FALSE)


peakAnno_H9_CRISPR_21_DE <- annotatePeak(dba_1D_CRISPR_21_DE, tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db", level = "gene")

df_H9_21 <- as.data.frame(peakAnno_H9_CRISPR_21_DE)
write.table(df_H9_21, file="peakAnno_H9_CRISPR_21_DiffBind.txt", sep="\t", col.names = TRUE, row.names = FALSE)


```

##Convert Diffbind object to GRanges object
gr <- makeGRangesFromDataFrame(df_1d_H1_vs_H9,keep.extra.columns = TRUE)  
gr <- keepStandardChromosomes(gr, pruning.mode="coarse") ##Remove dumb contigs
gr <- dropSeqlevels(gr,  c("chrX","chrY"), pruning.mode="coarse") ##Remove Sex Chomosomes
df <- as.data.frame(gr)

```{r setup, message=FALSE}
# Last modified 2018-08-19
bcbioRNASeq::prepareRNASeqTemplate()
source("_setup.R")

# Directory paths ==============================================================
invisible(mapply(
    FUN = dir.create,
    path = c(params$data_dir, params$results_dir),
    MoreArgs = list(showWarnings = FALSE, recursive = TRUE)
))

# Authenticate with Dropbox, if enabled
if (!is.null(params$dropbox_dir)) {
    rdrop2::drop_auth()
}

# Load object ==================================================================
bcb_name <- load(params$bcb_file)
bcb <- get(bcb_name, inherits = FALSE)
stopifnot(is(bcb, "bcbioRNASeq"))
invisible(validObject(bcb))
print(bcb)
```

```{r header, child="_header.Rmd"}
```

# Create `DESeqDataSet`

Here we are using an S4 coercion method to convert our `bcbioRNASeq` object to a `DESeqDataSet`. This prepares a gene-level `RangedSummarizedExperiment` with raw integer counts defined in the `assay()` slot. Internally this uses the `DESeqDataSet()` constructor function and sets an empty design formula. The desired design formula can be set with the `design()` function.

```{r coerce}
# Coerce bcbioRNASeq to DESeqDataSet
# help("coerce", "bcbioRNASeq")
# getMethod(
#     f = "coerce",
#     signature = signature(
#         from = "bcbioRNASeq",
#         to = "DESeqDataSet"
#     )
# )
dds <- as(bcb, "DESeqDataSet")
print(dds)
```

```{r design}
# colnames(colData(dds))
dds$group <- factor(paste0(dds$batch, dds$treatment))
design(dds) <- ~ group
```

```{r prefilter}
# Note that this criteria can be made more stringent
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep, ]
print(dds)
```

```{r deseq}
dds <- DESeq(dds)
saveData(dds, dir = params$data_dir)
```

```{r variance_stabilization}
# Alternatively, can use `rlog()` here, but it is slower
rld <- varianceStabilizingTransformation(dds)
interestingGroups(rld) <- "treatment"
saveData(rld, dir = params$data_dir)
```



```{r results}
res_H1_d14 <- results(dds, contrast=c("group","",""))
res_H1_d21 <- results(dds, contrast=c("group","",""))
res_H9_d14 <- results(dds, contrast=c("group","",""))
res_H9_d21 <- results(dds, contrast=c("group","",""))

```

# Results

For contrast argument as character vector:

1. Design matrix factor of interest.
2. Numerator for LFC (expt).
3. Denominator for LFC (control).

```{r contrasts}
# factor; numerator; denominator
# levels(dds$genotype)
# help("results", "DESeq2")
stopifnot(is.list(params$contrasts))
contrasts <- params$contrasts
names <- vapply(
    X = contrasts,
    FUN = function(x) {
        paste(x[[1]], x[[2]], "vs", x[[3]], sep = "_")
        
    },
    FUN.VALUE = "character"
)
names(contrasts) <- names
print(contrasts)
```

```{r res_unshrunken}
res_list_unshrunken <- mapply(
    FUN = results,
    contrast = contrasts,
    MoreArgs = list(
        object = dds,
        alpha = params$alpha
    ),
    SIMPLIFY = FALSE,
    USE.NAMES = FALSE
)
names(res_list_unshrunken) <- names
saveData(res_list_unshrunken, dir = params$data_dir)
```

Now let's calculate shrunken log2 fold change values with `DESeq2::lfcShrink()`. We're using the "normal" shrinkage estimator (default in DESeq2); the "apeglm" shrinkage estimator is also promising but doens't work well with complex contrast designs.

```{r res_shrunken}
# For `type` arguments other than "normal", `coef` argument is required.
# Otherwise can use `contrast`, which is more intuitive and readable.
# If using `coef` number, must match order in `resultsNames(dds)`.
# The new apeglm method is great but currently only works with coef.
# help("lfcShrink", "DESeq2")
# help("coef", "DESeq2")
# help("resultsNames", "DESeq2")
# This step can be a little slow and sped up with the `parallel` argument.
res_list_shrunken <- mapply(
    FUN = lfcShrink,
    res = res_list_unshrunken,
    contrast = contrasts,
    MoreArgs = list(
        dds = dds,
        type = "normal"
    ),
    SIMPLIFY = FALSE,
    USE.NAMES = TRUE
)
saveData(res_list_shrunken, dir = params$data_dir)
```

```{r plot_ma}
mapply(
    FUN = plotMA,
    object = res_list_shrunken,
    MoreArgs = list(lfcThreshold = params$lfc),
    SIMPLIFY = FALSE
)
```

## Volcano

A volcano plot compares significance (BH-adjusted *P* value) against fold change (log2) [@Cui2003-rn; @Li2014-ll]. Genes in the green box with text labels have an adjusted *P* value are likely to be the top candidate genes of interest.

```{r plot_volcano}
mapply(
    FUN = plotVolcano,
    object = res_list_shrunken,
    MoreArgs = list(lfcThreshold = params$lfc),
    SIMPLIFY = FALSE
)
```

## PCA

```{r plot_deg_pca}
# help("pheatmap", "pheatmap")
mapply(
    FUN = plotDEGPCA,
    results = res_list_shrunken,
    MoreArgs = list(
        counts = rld,
        label = TRUE
    ),
    SIMPLIFY = FALSE
)
```

## Heatmap

This plot shows only differentially expressed genes on a per-sample basis. We have scaled the data by row and used the `ward.D2` method for clustering [@Ward1963-xf].

```{r plot_deg_heatmap}
# help("pheatmap", "pheatmap")
invisible(mapply(
    FUN = plotDEGHeatmap,
    results = res_list_shrunken,
    MoreArgs = list(
        counts = rld,
        clusteringMethod = "ward.D2",
        scale = "row"
    ),
    SIMPLIFY = FALSE
))
```

# Results tables

Subset the results into separate tables, containing all genes, differentially expressed genes in both directions, and directional tables.

```{r results_tables, results="asis"}
# Here we're creating subset tables of the DEGs, and adding the normalized
# counts used by DESeq2 for the differential expression analysis.
res_tbl_list <- mapply(
    FUN = resultsTables,
    results = res_list_shrunken,
    MoreArgs = list(
        counts = dds,
        lfcThreshold = params$lfc,
        summary = TRUE,
        headerLevel = 2,
        write = TRUE,
        dir = params$results_dir,
        dropboxDir = params$dropbox_dir
    ),
    SIMPLIFY = FALSE,
    USE.NAMES = TRUE
)
saveData(res_tbl_list, dir = params$data_dir)
```

Differentially expressed gene (DEG) tables are sorted by BH-adjusted P value, and contain the following columns:

- `baseMean`: Mean of the normalized counts per gene for all samples.
- `log2FoldChange`: log2 fold change.
- `lfcSE`: log2 standard error.
- `stat`: Wald statistic.
- `pvalue`: Walt test *P* value.
- `padj`: BH adjusted Wald test *P* value (corrected for multiple comparisons; aka FDR).

## Top tables

Only the top up- and down-regulated genes (arranged by log2 fold change) are shown.

```{r top_tables, results="asis"}
invisible(mapply(
    FUN = topTables,
    object = res_tbl_list,
    MoreArgs = list(n = 20L, coding = FALSE)
))
```