---
title: "Repair_Seq_Analysis Figure 2"
output: html_notebook
params:
  # bcbioRNASeq object
  bcb_file: "repairseq_bcbV2.rda"
---

```{r setup, message=FALSE}
library("DiffBind")
library("ChIPpeakAnno")
library("ChIPseeker")
library("pqsfinder")
library("motifRG")
library("BSgenome.Hsapiens.UCSC.hg38")
library("EnsDb.Hsapiens.v86")
library("TxDb.Hsapiens.UCSC.hg38.knownGene")
library("clusterProfiler")
library("ReactomePA")
library("org.Hs.eg.db")
library("viridis")
library("ggplot2")
library("ggExtra")
library("ggpmisc")
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
anno86 <- annoGR(EnsDb.Hsapiens.v86)
# Last modified 2018-08-19
bcbioRNASeq::prepareRNASeqTemplate()
source("_setup.R")

# Load object ==================================================================
repair_bcb_name <- load(params$bcb_file)
repair_bcb <- get(repair_bcb_name, inherits = FALSE)
stopifnot(is(repair_bcb, "bcbioRNASeq"))
invisible(validObject(repair_bcb))
print(repair_bcb)
```

```{r sample_data}
# getMethod("sampleData", "SummarizedExperiment")
sample_data <- sampleData(repair_bcb, clean = TRUE) %>% as.data.frame()
write.csv(
    x = sample_data,
    file = file.path("Repiar_BCB_sample_data.csv")
)
sample_data
```

```{r counts}
# Raw counts (don't use for plotting)
repair_raw_counts <- counts(repair_bcb, normalized = FALSE)
# DESeq2 normalized counts
repair_normalized_counts <- counts(repair_bcb, normalized = TRUE)
# Transcripts per million
repair_tpm <- counts(repair_bcb, normalized = "tpm")
saveData(repair_raw_counts, repair_normalized_counts, repair_tpm, dir = ".")
writeCounts(repair_raw_counts, repair_normalized_counts, repair_tpm, dir = ".")

```

```{r}


# Fig. 2. Relationship between DNA repair and transcription in human neurons.

# (A) RPM (repairs per kilobase million) from genome repair hotspots compared with TPM (transcripts per kilobase million) from total RNA-Seq define 4 distinct classes of genes based on their repair and transcription.

# (C) Gene ontology analysis for DNA repair hotspots.

# (E) Box plots for eQTLs in promoters/intergenic regions.

```


```{r}
#(A) RPM (repairs per kilobase million) from genome repair hotspots compared with TPM (transcripts per kilobase million) from total RNA-Seq define 4 distinct classes of genes based on their repair and transcription.



# (B) Total RPMs compared with TPMs demonstrate few genes change classes when accounting for all measured repairs.

load("repair_tpm.rda")
load("RNA_tpm.rda")


RNA_tpm_df <- as.data.frame(RNA_tpm)
Repair_tpm_df <- as.data.frame(repair_tpm)

RNA_TPM_H1H9 <- RNA_tpm_df[,c(1,4,5,10)]
RNA_TPM_H1H9$ENSEMBL <- rownames(RNA_TPM_H1H9)
RNA_TPM_H1H9$avg_RNA_TPM <- rowMeans(RNA_TPM_H1H9[c('H9UNG_A', 'H9UNG_B','H9UNG_A', 'H9UNG_B')], na.rm=FALSE)

Repair_TPM_H1H9 <- Repair_tpm_df[,c(27,28,77,78)]
Repair_TPM_H1H9$ENSEMBL <- rownames(Repair_TPM_H1H9)
Repair_TPM_H1H9$avg_Repair_TPM <- rowMeans(Repair_TPM_H1H9[c('H1UNG_EdU_1d_A', 'H1UNG_EdU_1d_B','H9UNG_EdU_1d_A', 'H9UNG_EdU_1d_B')], na.rm=FALSE)


H1H9_1D_TPM_Data <- merge(RNA_TPM_H1H9, Repair_TPM_H1H9, by.x="ENSEMBL", by.y ="ENSEMBL", all=TRUE)

H1H9_1D_TPM_Data$Class[H1H9_1D_TPM_Data$avg_RNA_TPM > 0 & H1H9_1D_TPM_Data$avg_Repair_TPM > 0] <- "Class_4"
H1H9_1D_TPM_Data$Class[H1H9_1D_TPM_Data$avg_RNA_TPM == 0 & H1H9_1D_TPM_Data$avg_Repair_TPM > 0] <- "Class_3"
H1H9_1D_TPM_Data$Class[H1H9_1D_TPM_Data$avg_RNA_TPM > 0 & H1H9_1D_TPM_Data$avg_Repair_TPM==0] <- "Class_2"
H1H9_1D_TPM_Data$Class[H1H9_1D_TPM_Data$avg_RNA_TPM == 0 & H1H9_1D_TPM_Data$avg_Repair_TPM==0] <- "Class_1"

# (B) Total RPMs compared with TPMs demonstrate few genes change classes when accounting for all measured repairs.
ggplot(H1H9_1D_TPM_Data, aes(x=avg_RNA_TPM + .0001, y=avg_Repair_TPM + .0001))  + scale_x_log10() + scale_y_log10() +
geom_point(alpha = 0.2, size=.5, aes(colour=Class))+
xlab("RNA TPMs")+
ylab("Repair TPMs")+
theme_classic()  + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))


```