---
title: "Repair_Seq_Analysis"
output: html_notebook
---
```{r}
#Table S2. List of Class 1-4 Genes.
```

```{r}
library("DiffBind")
library("ChIPpeakAnno")
library("ChIPseeker")
library("pqsfinder")
library("motifRG")
library("BSgenome.Hsapiens.UCSC.hg38")
library("EnsDb.Hsapiens.v86")
library("TxDb.Hsapiens.UCSC.hg38.knownGene")
library("clusterProfiler")
library("ReactomePA")
library("org.Hs.eg.db")
library("viridis")
library("ggplot2")
library("ggExtra")
library("ggpmisc")



txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
anno86 <- annoGR(EnsDb.Hsapiens.v86)

```

```{r}
### Load new dba and bcb objects.
load("repairseq_dbaV2.rdata")
load("repairseq_bcbV2.rda")

# New masks basic 1D Peaks Common across H1 and H9
dba_H1H9_mask <- dba.mask(DBAobject,DBA_ID, c("H1","H9"))
dba_EdU_mask <- dba.mask(DBAobject,DBA_CONDITION, "EdU", mask=dba_H1H9_mask, merge="and")
dba_EdU_1Day_mask <- dba.mask(DBAobject,DBA_FACTOR,"1Day",mask=dba_EdU_mask, merge="and")
dba_EdU_1Day_None_mask <- dba.mask(DBAobject,DBA_TREATMENT,"None",mask=dba_EdU_1Day_mask, merge="and")
dba_EdU_1Day_None_ESCin_mask <- dba.mask(DBAobject,DBA_TISSUE,"ESC-iN",mask=dba_EdU_1Day_None_mask, merge="and")
dba_1Day_EdU_None_ESCin <- dba(DBAobject,dba_EdU_1Day_None_ESCin_mask)
dba_1D <- dba.contrast(dba_1Day_EdU_None_ESCin, categories=DBA_ID, minMembers = 2)
dba_1D <- dba.analyze(dba_1D, method=DBA_DESEQ2)
dba_1D_Report <- dba.report(dba_1D, contrast=1, th=1, bUsePval=TRUE)
dba_1D_Common <- dba_1D_Report[dba_1D_Report$`p-value` > .05]

#Annotate 1D peaks and plot genomic distribition
peakAnno_1D_Common <- annotatePeak(dba_1D_Common, tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db")
plotAnnoPie(peakAnno_1D_Common) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))

#Look at functional enrichemnt of genes with 1D peaks
gene <- seq2gene(dba_1D_Common, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb, sameStrand=FALSE)
ego <- enrichGO(gene=gene, OrgDb=org.Hs.eg.db, ont="BP", pvalueCutoff =0.05)
dotplot(ego) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
cnetplot(ego, categorySize="pvalue", foldChange=gene)

##H1 vs H9 correlation


df_1d_H1_vs_H9 <- as.data.frame(dba_1D_Report)
##Convert Diffbind object to GRanges object
gr <- makeGRangesFromDataFrame(df_1d_H1_vs_H9,keep.extra.columns = TRUE)  
gr <- keepStandardChromosomes(gr, pruning.mode="coarse") ##Remove dumb contigs
gr <- dropSeqlevels(gr,  c("chrX","chrY"), pruning.mode="coarse") ##Remove Sex Chomosomes
df <- as.data.frame(gr)
my.formula <- y ~ x
ggplot(df, aes(x=Conc_H1, y=Conc_H9)) + geom_point(size=.2) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + stat_poly_eq(formula = my.formula, aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), parse = TRUE) + xlim(2,10)  + ylim(2,10)

## KCL KCL low KCL+pki pki and low TTX == Activity mask

## 1D edu 1D edc 1d edu+edc and GQ stabalizer == eduedc mask

## lowH202 NCS ETP TOP1 and TOP2 and ParPi

## 1D 4D 7D peaks no treatment

### mpa bcb tpms to dba object




```


```{r}
## Condtion based comparison to define peaks common to 1D 4D and 7D
dbaset_1D4D7D <- dba.contrast(dbaset, categories=DBA_CONDITION)  
dbaset_1D4D7D <- dba.analyze(dbaset_1D4D7D, method=DBA_DESEQ2)
```

```{r}
#1D vs 4D
dbaset_1Dvs4D <- dba.report(dbaset_1D4D7D, contrast=1, th=1, bUsePval=TRUE)
dba_1D4D_Common <- dbaset_1Dvs4D[dbaset_1Dvs4D$`p-value` > .05]
#1D vs 7D
dbaset_1Dvs7D <- dba.report(dbaset_1D4D7D, contrast=2, th=1, bUsePval=TRUE)
dba_1D7D_Common <- dbaset_1Dvs7D[dbaset_1Dvs7D$`p-value` > .05]
#4D vs 7D
dbaset_4Dvs7D <- dba.report(dbaset_1D4D7D, contrast=3, th=1, bUsePval=TRUE)
dba_4D7D_Common <- dbaset_4Dvs7D[dbaset_4Dvs7D$`p-value` > .05]

dbaset_1D4D7D_Common_tmp <- intersect(dba_1D4D_Common, dba_1D7D_Common, ignore.strand=TRUE)
dbaset_1D4D7D_Common <- intersect(dbaset_1D4D7D_Common_tmp, dba_4D7D_Common, ignore.strand=TRUE)

pdf("1D4D7D_Common_RepairSeq_Peak_Figs.pdf")
peakAnno_Common <- annotatePeak(dbaset_1D4D7D_Common, tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db")
plotAnnoPie(peakAnno_Common) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
upsetplot(peakAnno_Common, vennpie=FALSE) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
vennpie(peakAnno_Common) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
gene <- seq2gene(dbaset_1D4D7D_Common, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb, sameStrand=FALSE)
ego <- enrichGO(gene=gene, OrgDb=org.Hs.eg.db, ont="BP", pvalueCutoff =0.05)
dotplot(ego) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
dev.off()
```

```{r}
makeVennDiagram(list(dba_1D4D_Common, dba_1D7D_Common, dba_4D7D_Common), NameOfPeaks=c("Common_1D4D", "Common_1D7D","Common_4D7D"))

```

```{r}
## for motifs and pqsfinder 
peaksWithSequences <- getAllPeakSequence(dbaset_1D4D7D_Common, upstream=0, downstream=0, genome=Hsapiens)
write2FASTA(peaksWithSequences, "RepairSeq_CommonPeaks.fa")

```

```{r}
cnetplot(ego, categorySize="pvalue", foldChange=gene)
```

```{r}
#1 Day Specific Peak Analysis
dbaset_1D_Specific <- dba.contrast(dbaset, dbaset$masks$`1dEdU`, bNot = TRUE)
dbaset_1D_Specific <- dba.analyze(dbaset_1D_Specific, method=DBA_DESEQ2)
dba_1D_report <- dba.report(dbaset_1D_Specific, th=.05, bUsePval=TRUE, fold=2)
peakAnno_1D <- annotatePeak(dba_1D_report[dba_1D_report$Fold >2], tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db")
pdf("1D_Specific_RepairSeq_Peak_Figs.pdf")
plotAnnoPie(peakAnno_1D) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
upsetplot(peakAnno_1D, vennpie=FALSE) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
vennpie(peakAnno_1D) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
gene <- seq2gene(dba_1D_report[dba_1D_report$Fold >2], tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb, sameStrand=FALSE)
ego <- enrichGO(gene=gene, OrgDb=org.Hs.eg.db, ont="BP", pvalueCutoff =0.05)
dotplot(ego) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
dev.off()
```



```{r}
#4 day speciifc peak analysis
dbaset_4D_Specific <- dba.contrast(dbaset, dbaset$masks$`4dEdU`, bNot = TRUE)
dbaset_4D_Specific <- dba.analyze(dbaset_4D_Specific, method=DBA_DESEQ2)
dba_4D_report <- dba.report(dbaset_4D_Specific, th=.05, bUsePval=TRUE, fold=2)
peakAnno_4D <- annotatePeak(dba_4D_report[dba_4D_report$Fold >2], tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db")
plotAnnoPie(peakAnno_4D) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
upsetplot(peakAnno_4D, vennpie=FALSE) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
gene <- seq2gene(dba_4D_report[dba_4D_report$Fold >2], tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb, sameStrand=FALSE)
ego <- enrichGO(gene=gene, OrgDb=org.Hs.eg.db, ont="BP", pvalueCutoff =0.05)
dotplot(ego) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
```

```{r}
#7 day specific peak analysis
dbaset_7D_Specific <- dba.contrast(dbaset, dbaset$masks$`7dEdU`, bNot = TRUE)
dbaset_7D_Specific <- dba.analyze(dbaset_7D_Specific, method=DBA_DESEQ2)
dba_7D_report <- dba.report(dbaset_7D_Specific, th=.05, bUsePval=TRUE, fold=2)
peakAnno_7D <- annotatePeak(dba_7D_report[dba_7D_report$Fold >2], tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db")
plotAnnoPie(peakAnno_7D) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
upsetplot(peakAnno_7D, vennpie=FALSE) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
gene <- seq2gene(dba_7D_report[dba_7D_report$Fold >2], tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb, sameStrand=FALSE)
ego <- enrichGO(gene=gene, OrgDb=org.Hs.eg.db, ont="BP", pvalueCutoff =0.05)
dotplot(ego) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
```

```{r}
#4Day 7Day Specific Peak Analysis
#dbaset_1D_Specific <- dba.contrast(dbaset, dbaset$masks$`1dEdU`, bNot = TRUE)
#dbaset_1D_Specific <- dba.analyze(dbaset_1D_Specific, method=DBA_DESEQ2)
dba_1D_report <- dba.report(dbaset_1D_Specific, th=.05, bUsePval=TRUE, fold=2)
peakAnno_4D7D <- annotatePeak(dba_1D_report[dba_1D_report$Fold <2], tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db")
pdf("4D7D_Specific_RepairSeq_Peak_Figs.pdf")
plotAnnoPie(peakAnno_4D7D) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
upsetplot(peakAnno_4D7D, vennpie=FALSE) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
vennpie(peakAnno_4D7D) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
gene <- seq2gene(dba_1D_report[dba_1D_report$Fold <2], tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb, sameStrand=FALSE)
ego <- enrichGO(gene=gene, OrgDb=org.Hs.eg.db, ont="BP", pvalueCutoff =0.05)
dotplot(ego) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
dev.off()
```

```{r}
#1D 4D 7D Pairwise corelations

df_1dvs4d <- as.data.frame(dbaset_1Dvs4D)
df_1dvs7d <- as.data.frame(dbaset_1Dvs7D)
df_4dvs7d <- as.data.frame(dbaset_4Dvs7D)
my.formula <- y ~ x
pdf("1D4D7D_RepairSeq_correlation_scatter.pdf")
ggplot(df_1dvs4d, aes(x=Conc_1dEdU, y=Conc_4dEdU)) + geom_point(size=.2) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + stat_poly_eq(formula = my.formula, aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), parse = TRUE)         

ggplot(df_1dvs7d, aes(x=Conc_1dEdU, y=Conc_7dEdU)) + geom_point(size=.2) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + stat_poly_eq(formula = my.formula, aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), parse = TRUE) 

ggplot(df_4dvs7d, aes(x=Conc_4dEdU, y=Conc_7dEdU)) + geom_point(size=.2) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + stat_poly_eq(formula = my.formula, aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), parse = TRUE) 
dev.off()
```

```{r}
ggplot(as.data.frame(dba_1D4D7D_Peaks)) + geom_point(size = 1, aes(x=dba_1D4D7D_Peaks$RPKM, y=dba_1D4D7D_Peaks$Score))
ggplot(as.data.frame(dba_1D4D7D_Peaks)) + geom_point(size = 1, aes(x=dba_1D4D7D_Peaks$Reads, y=dba_1D4D7D_Peaks$Score))
ggplot(as.data.frame(dba_1D4D7D_Peaks)) + geom_point(size = 1, aes(x=dba_1D4D7D_Peaks$cRPKM, y=dba_1D4D7D_Peaks$Score))
ggplot(as.data.frame(dba_1D4D7D_Peaks)) + geom_point(size = 1, aes(x=dba_1D4D7D_Peaks$cReads, y=dba_1D4D7D_Peaks$Score))
ggplot(as.data.frame(X)) + geom_point(size = 1, aes(x=X$Reads, y=X$ScoreNew))

```


```{r}

setwd("~/RepairSeq/dba_repair_1D4D7D/")
files <- as.list(list.files(pattern = "\\.narrowPeak$"))
gr <- makeGRangesFromDataFrame(as.data.frame(dbaset$peaks),keep.extra.columns = TRUE)  ##Convert Diffbind object to GRanges object
gr <- keepStandardChromosomes(gr, pruning.mode="coarse") ##Remove dumb contigs
gr <- dropSeqlevels(gr,  c("chrX","chrY"), pruning.mode="coarse") ##Remove Sex Chomosomes

peakList <- lapply(files, readPeakFile)
peakList_Standard <- lapply(peakList, keepStandardChromosomes, pruning.mode="coarse")
peakList_Autosome <- lapply(peakList_Standard, dropSeqlevels, c("chrX","chrY"), pruning.mode="coarse")
peakAnnoList <- lapply(peakList_Autosome, annotatePeak, TxDb=txdb,tssRegion=c(-3000, 3000), verbose=TRUE)
pdf("1D4D7D_FeatureDistribution.pdf") 
plotAnnoBar(peakAnnoList) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
dev.off() 
```

```{r}

```


```{r}
peakAnno <- annotatePeak(gr_full, tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db")
plotAnnoBar(peakAnno) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
```


```{r}
plotAnnoBar(gr) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
```

```{r}
pdf("1D4D7D_UpsetPlot.pdf", width=14) 
upsetplot(peakAnno, vennpie=TRUE) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
dev.off()
```

```{r}
pathway1 <- enrichPathway(as.data.frame(peakAnno)$geneId)
gene <- seq2gene(gr_auto, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb, sameStrand=FALSE)
pathway2 <- enrichPathway(gene)
dotplot(pathway2) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
```

```{r}
ego <- enrichGO(gene=gene, OrgDb=org.Hs.eg.db, ont="BP", pvalueCutoff =0.05)
```

```{r}
pdf("1D4D7D_GO_TermEnrichment.pdf") 
dotplot(ego) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
dev.off()
```


```{r}

```


```{r}
setwd("~/RepairSeq/dba_repair_1D4D7D/")
a = list.files(pattern = ("*genic\\.narrowPeak$"))
peakList <- lapply(a, readPeakFile)
peakList_Standard <- lapply(peakList, keepStandardChromosomes, pruning.mode="coarse")
peakList_Autosome <- lapply(peakList_Standard, dropSeqlevels, c("chrX","chrY"), pruning.mode="coarse")
peakAnnoList <- lapply(peakList_Autosome, annotatePeak, TxDb=txdb,tssRegion=c(-3000, 3000), verbose=TRUE)
plotAnnoBar(peakAnnoList)
```

```{r}
setwd("~/RepairSeq/dba_samples_Activity/")
files <- as.list(list.files(pattern = "\\.narrowPeak$"))
peakList <- lapply(files, readPeakFile)
peakList_Standard <- lapply(peakList, keepStandardChromosomes, pruning.mode="coarse")
peakList_Autosome <- lapply(peakList_Standard, dropSeqlevels, c("chrX","chrY"), pruning.mode="coarse")
peakAnnoList <- lapply(peakList_Autosome, annotatePeak, TxDb=txdb,tssRegion=c(-3000, 3000), verbose=TRUE)
plotAnnoBar(peakAnnoList)
```

```{r}
setwd("~/RepairSeq/dba_samples_NoTreatment//")
files <- as.list(list.files(pattern = "\\.narrowPeak$"))
peakList <- lapply(files, readPeakFile)
peakList_Standard <- lapply(peakList, keepStandardChromosomes, pruning.mode="coarse")
peakList_Autosome <- lapply(peakList_Standard, dropSeqlevels, c("chrX","chrY"), pruning.mode="coarse")
peakAnnoList <- lapply(peakList_Autosome, annotatePeak, TxDb=txdb,tssRegion=c(-3000, 3000), verbose=TRUE)
plotAnnoBar(peakAnnoList)
```

```{r}
#peakAnno <- annotatePeak(gr,tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="EnsDb.Hsapiens.v86")  #annotation of all peaks

NoTreatment_Files <- as.list(list.files(path="~/RepairSeq/dba_samples_NoTreatment/", pattern = "\\.narrowPeak$"))
print(NoTreatment_Files)
peakAnnoList <- lapply(NoTreatment_Files, annotatePeak, TxDb=txdb,tssRegion=c(-3000, 3000), verbose=FALSE)
plotAnnoBar(peakAnnoList)


peak <- keepStandardChromosomes(peak, pruning.mode="coarse") ##Remove dumb contigs


tagHeatmap(tagMatrixList, xlim=c(-3000, 3000), color=NULL)
View(files)

files <- as.list(list.files())
files <- setNames(files,files)
```


```{r}
promoter <- getPromoters(TxDb=txdb, upstream=3000, downstream=3000)
tagMatrixList <- lapply(files, getTagMatrix, windows=promoter)
tagHeatmap(tagMatrixList[1:5], xlim=c(-3000, 3000), color=NULL)

#gra = annotatePeakInBatch(gr, AnnotationData=anno86)
#covplot(gr)
```
```{r}
setwd("~/RepairSeq/dba_samples_Activity//")
files <- as.list(list.files(pattern = "\\.narrowPeak$"))
peakAnno <- annotatePeak(files[[1]], tssRegion=c(-3000, 3000),TxDb=txdb)
pathway1 <- enrichPathway(as.data.frame(peakAnno)$geneId)
peak <- readPeakFile(files[[1]])
gene <- seq2gene(peak, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb)
pathway2 <- enrichPathway(gene)
dotplot(pathway2) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
```

```{r}
setwd("~/RepairSeq/dba_samples_Activity/")
files <- as.list(list.files(pattern = "\\.narrowPeak$"))
peakAnno <- annotatePeak(files[[7]], tssRegion=c(-3000, 3000),TxDb=txdb)
pathway1 <- enrichPathway(as.data.frame(peakAnno)$geneId)
peak <- readPeakFile(files[[7]])
gene <- seq2gene(peak, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb)
pathway2 <- enrichPathway(gene)
dotplot(pathway2)
```

```{r}
setwd("~/RepairSeq/dba_samples_Activity/")
files <- as.list(list.files(pattern = "\\.narrowPeak$"))
peakAnno <- annotatePeak(files[[13]], tssRegion=c(-3000, 3000),TxDb=txdb)
pathway1 <- enrichPathway(as.data.frame(peakAnno)$geneId)
peak <- readPeakFile(files[[13]])
gene <- seq2gene(peak, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb)
pathway2 <- enrichPathway(gene)
dotplot(pathway2)
```

```{r}
setwd("~/RepairSeq/dba_samples_Activity/")
files <- as.list(list.files(pattern = "\\.narrowPeak$"))
peakAnno <- annotatePeak(files[[14]], tssRegion=c(-3000, 3000),TxDb=txdb)
pathway1 <- enrichPathway(as.data.frame(peakAnno)$geneId)
peak <- readPeakFile(files[[14]])
gene <- seq2gene(peak, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb)
pathway2 <- enrichPathway(gene)
dotplot(pathway2)
```

```{r}
setwd("~/RepairSeq/dba_samples_Activity/")
files <- as.list(list.files(pattern = "\\.narrowPeak$"))
peakAnno <- annotatePeak(files[[15]], tssRegion=c(-3000, 3000),TxDb=txdb)
pathway1 <- enrichPathway(as.data.frame(peakAnno)$geneId)
peak <- readPeakFile(files[[15]])
gene <- seq2gene(peak, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb)
pathway2 <- enrichPathway(gene)
dotplot(pathway2)
```


```{r}
#setwd("~/RepairSeq/dba_samples_NoTreatment//")
#files <- as.list(list.files(pattern = "\\.narrowPeak$"))
#peakList <- lapply(files, readPeakFile)
#peakList_Standard <- lapply(peakList, keepStandardChromosomes, pruning.mode="coarse")
#peakList_Autosome <- lapply(peakList_Standard, dropSeqlevels, c("chrX","chrY"), pruning.mode="coarse")
#peakAnnoList <- lapply(peakList_Autosome, annotatePeak, TxDb=txdb,tssRegion=c(-3000, 3000), verbose=TRUE)
genes = lapply(peakAnnoList_Small, function(i) as.data.frame(i)$geneId)
names(genes) = sub("_", "\n", names(genes))
compKEGG <- compareCluster(geneCluster= genes,fun= "enrichKEGG",pvalueCutoff  = 0.05,pAdjustMethod = "BH")
dotplot(compKEGG, showCategory = 15, title = "KEGG Pathway Enrichment Analysis")
```

```{r}
setwd("~/RepairSeq/dba_samples_AllRepair/")
files <- as.list(list.files(pattern = "\\.narrowPeak$"))
peakAnno <- annotatePeak(files[[3]], tssRegion=c(-3000, 3000),TxDb=txdb)
pathway1 <- enrichPathway(as.data.frame(peakAnno)$geneId)
peak <- readPeakFile(files[[3]])
gene <- seq2gene(peak, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb)
pathway2 <- enrichPathway(gene)
dotplot(pathway2) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))
```

```{r}
setwd("~/RepairSeq/dba_samples_AllRepair/")
files <- as.list(list.files(pattern = "\\.narrowPeak$"))
peakAnno <- annotatePeak(files[[5]], tssRegion=c(-3000, 3000),TxDb=txdb)
pathway1 <- enrichPathway(as.data.frame(peakAnno)$geneId)
peak <- readPeakFile(files[[5]])
gene <- seq2gene(peak, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb)
pathway2 <- enrichPathway(gene)
dotplot(pathway2) + theme_classic()
```

```{r}
genes = lapply(peakAnnoList, function(i) as.data.frame(i)$geneId)
names(genes) = sub("_", "\n", names(genes))
compKEGG <- compareCluster(geneCluster= genes, fun= "enrichKEGG", pvalueCutoff  = 0.05, pAdjustMethod = "BH")
dotplot(compKEGG, showCategory = 15, title = "KEGG Pathway Enrichment Analysis")
```

```{r}
dotplot(pathway2) + theme_classic()
```

```{r}
seq <-getSequence(test, genome=BSgenome.Hsapiens.UCSC.hg38)
```

```{r}
seq[500]
pqsfinder(seq[[500]])
```

```{r}
for (i in 1:length(seq)){
  seq@metadata[i] <-  pqsfinder(seq[[i]])
}
```


```{r}
#Profile of ChIP peaks binding to TSS regions
promoter <- getPromoters(TxDb=txdb, upstream=3000, downstream=3000)
tagMatrixList <- lapply(files, getTagMatrix, windows=promoter)
tagHeatmap(tagMatrixList, xlim=c(-3000, 3000), color=NULL)
```


```{r}
##shuffle within all genes in trascroptome
setwd("~/RepairSeq/dba_repair_1D4D7D")
load("~/RepairSeq/dba_repair_1D4D7D/dbaset.rda")
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
anno86 <- annoGR(EnsDb.Hsapiens.v86)

dba_1D4D7D_Peaks <- makeGRangesFromDataFrame(as.data.frame(dbaset$peaks),keep.extra.columns = TRUE)
dba_1D4D7D_df <- as.data.frame(dba_1D4D7D_Peaks)
peakAnno_1D4D7D <- annotatePeak(dba_shuffle, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")
#write.table(dba_1D4D7D_df, file="dba_1D4D7D_All.bed", sep="\t", col.names = FALSE, row.names = FALSE)

#####Create genic subset and shuffle with bedtools then load back into R##############

library(readr)
setwd("~/RepairSeq/dba_repair_1D4D7D")
dba_1D4D7D_Genic <- read_delim("dba_1D4D7D_Genic.bed", "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
dba_1D4D7D_Shuffle <- read_delim("repair_genes_shuffle_new.bed", "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
#dba_1D4D7D_Shuffle2 <- read_delim("dba_1D4D7D_Shuffle2.bed", "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)

colnames(dba_1D4D7D_Genic) <- colnames(dba_1D4D7D_df)
colnames(dba_1D4D7D_Shuffle) <- colnames(dba_1D4D7D_df)
#colnames(dba_1D4D7D_Shuffle2) <- colnames(dba_1D4D7D_df)


dba_1D4D7D_Genic_gr <- makeGRangesFromDataFrame(as.data.frame(dba_1D4D7D_Genic),keep.extra.columns = TRUE)
dba_1D4D7D_Shuffle_gr <- makeGRangesFromDataFrame(as.data.frame(dba_1D4D7D_Shuffle),keep.extra.columns = TRUE)
#dba_1D4D7D_Shuffle2_gr <- makeGRangesFromDataFrame(as.data.frame(dba_1D4D7D_Shuffle2),keep.extra.columns = TRUE)


peakAnno_1D4D7D_Genic <- annotatePeak(dba_1D4D7D_Genic_gr, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")
peakAnno_1D4D7D_Shuffle <- annotatePeak(dba_1D4D7D_Shuffle_gr, level="gene", annoDb="org.Hs.eg.db")
#peakAnno_1D4D7D_Shuffle2 <- annotatePeak(dba_1D4D7D_Shuffle2_gr, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")


genes_genic <- as.character(unique(peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL[peakAnno_1D4D7D@detailGenomicAnnotation$Intergenic == FALSE]))

#peakAnno_1D4D7D_All <- annotatePeak(dba_1D4D7D_Peaks, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")
genes_all <- as.character(unique(peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL))
```


```{r}
RepairData_Genic <- data.frame(genes_genic)
for(i in RepairData_Genic$genes_genic){
   RepairData_Genic$length[RepairData_Genic$genes_genic == i] <- max(peakAnno_1D4D7D@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)
}

RepairData_Genic <- RepairData_Genic[is.finite(RepairData_Genic$length),]
for(i in RepairData_Genic$genes_genic){
    RepairData_Genic$AverageScore[RepairData_Genic$genes_genic == i] <- mean(sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Score[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Score.1[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Score.2[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Score.3[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Score.4[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Score.5[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Score.6[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Score.7[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Score.8[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Score.9[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Score.10[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Score.11[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE))

    RepairData_Genic$AverageRPK[RepairData_Genic$genes_genic == i] <- mean(sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Reads[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Reads.1[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Reads.2[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Reads.3[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Reads.4[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Reads.5[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Reads.6[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Reads.7[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Reads.8[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Reads.9[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Reads.10[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D@anno@elementMetadata@listData$Reads.11[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE))/ (max(peakAnno_1D4D7D@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)/1000)
}

RepairData_Genic <- RepairData_Genic[complete.cases(RepairData_Genic),]
ScaleFactor <- sum(RepairData_Genic$AverageRPK) / 1000000
RepairData_Genic$AverageTPM <- RepairData_Genic$AverageRPK / ScaleFactor
quartiles <- quantile(RepairData_Genic$AverageTPM)
iqr <- IQR(RepairData_Genic$AverageTPM)
anomLimit <- as.numeric(quartiles[4]) + iqr*1.5

min <- min(RepairData_Genic$AverageTPM)
max <- max(RepairData_Genic$AverageTPM)

ggplot(RepairData_Genic) + geom_histogram(aes(x=AverageTPM), bins=1000) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + scale_x_log10()


RepairData_Genic$Class <- ifelse(RepairData_Genic$AverageTPM > anomLimit, "Guarded","Normal")
#RepairData_Genic$Class[RepairData_Genic$AverageTPM < anomLimit & RepairData_Genic$AverageTPM > quartiles[4]] <- "Guarded"
RepairData_Genic$Class[RepairData_Genic$AverageTPM <= quartiles[2]] <- "Unguarded"


p <- ggplot(RepairData_Genic, aes(length, AverageScore))+
  geom_point(alpha = 0.5, size=.25)+
  xlab("gene length")+
  ylab("Signal")+
  labs() + theme_classic() + scale_x_log10()  + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))  + ylim(0,100000)

ggMarginal(p, type="boxplot")

```

```{r}
##New randomized appraoch
peakAnno_1D4D7D_Random <- peakAnno_1D4D7D
peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL  <- sample(peakAnno_1D4D7D@anno@elementMetadata@listData$SYMBOL)

genes_Random <- as.character(unique(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL[peakAnno_1D4D7D_Random@detailGenomicAnnotation$Intergenic == FALSE]))
RepairData_Random <- data.frame(genes_Random)
for(i in RepairData_Random$genes_Random){
   RepairData_Random$length[RepairData_Random$genes_Random == i] <- max(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)
}

RepairData_Random <- RepairData_Random[is.finite(RepairData_Random$length),]
for(i in RepairData_Random$genes_Random){
    RepairData_Random$AverageScore[RepairData_Random$genes_Random == i] <- mean(sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Score[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Score.1[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Score.2[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Score.3[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Score.4[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Score.5[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Score.6[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Score.7[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Score.8[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Score.9[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Score.10[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Score.11[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE))

    RepairData_Random$AverageRPK[RepairData_Random$genes_Random == i] <- mean(sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Reads[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Reads.1[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Reads.2[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Reads.3[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Reads.4[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Reads.5[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Reads.6[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Reads.7[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Reads.8[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Reads.9[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Reads.10[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$Reads.11[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i & peakAnno_1D4D7D_Random@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE))/ (max(peakAnno_1D4D7D_Random@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D_Random@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)/1000)
}
RepairData_Random <- RepairData_Random[complete.cases(RepairData_Random),]
ScaleFactor <- sum(RepairData_Random$AverageRPK) / 1000000
RepairData_Random$AverageTPM <- RepairData_Random$AverageRPK / ScaleFactor
quartiles <- quantile(RepairData_Random$AverageTPM)
iqr <- IQR(RepairData_Random$AverageTPM)
anomLimit <- as.numeric(quartiles[4]) + iqr*1.5

min <- min(RepairData_Random$AverageTPM)
max <- max(RepairData_Random$AverageTPM)

ggplot(RepairData_Random) + geom_histogram(aes(x=AverageTPM), bins=1000) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + scale_x_log10()


RepairData_Random$Class <- ifelse(RepairData_Random$AverageTPM > anomLimit, "Guarded","Normal")
#RepairData_shuffle$Class[RepairData_shuffle$AverageTPM < anomLimit & RepairData_shuffle$AverageTPM > quartiles[4]] <- "Guarded"
RepairData_Random$Class[RepairData_Random$AverageTPM <= quartiles[2]] <- "Unguarded"


p2 <- ggplot(RepairData_Random, aes(length, AverageScore))+
  geom_point(alpha = 0.5, size=.25)+
  xlab("gene length")+
  ylab("Signal")+
  labs() + theme_classic() + scale_x_log10()  + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + ylim(0,100000)

ggMarginal(p2, type="boxplot")

```



```{r}
Unguarded_Genes_Genic <- as.character(RepairData_Genic[RepairData_Genic$Class == "Unguarded","genes_genic"])
Guarded_Genes_Genic <- as.character(RepairData_Genic[RepairData_Genic$Class == "Guarded","genes_genic"])
Normal_Genes_Genic <- as.character(RepairData_Genic[RepairData_Genic$Class == "Normal","genes_genic"])
Anomalous_Genes_Genic <- as.character(RepairData_Genic[RepairData_Genic$Class == "Anomalies","genes_genic"])
write.csv(Unguarded_Genes_Genic, file="Unguarded_Genes_Genic.csv")
write.csv(Guarded_Genes_Genic, file="Guarded_Genes_Genic.csv")
write.csv(Normal_Genes_Genic, file="Normal_Genes_Genic.csv")
write.csv(Anomalous_Genes_Genic, file="Anomalous_Genes_Genic.csv")
```

```{r}
dba_1D4D7D_Peaks <- makeGRangesFromDataFrame(as.data.frame(dbaset$peaks),keep.extra.columns = TRUE)
dba_shuffle <- shuffle(dba_1D4D7D_Peaks, TxDb=txdb)
dba_shuffle@elementMetadata <- dba_1D4D7D_Peaks@elementMetadata
peakAnno_1D4D7D_shuffle <- annotatePeak(dba_shuffle, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")

genes_shuffle <- as.character(unique(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL[peakAnno_1D4D7D_shuffle@detailGenomicAnnotation$Intergenic == FALSE]))
RepairData_shuffle <- data.frame(genes_shuffle)
for(i in RepairData_shuffle$genes_shuffle){
   RepairData_shuffle$length[RepairData_shuffle$genes_shuffle == i] <- max(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)
}

RepairData_shuffle <- RepairData_shuffle[is.finite(RepairData_shuffle$length),]
for(i in RepairData_shuffle$genes_shuffle){
    RepairData_shuffle$AverageScore[RepairData_shuffle$genes_shuffle == i] <- mean(sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Score[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Score.1[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Score.2[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Score.3[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Score.4[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Score.5[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Score.6[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Score.7[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Score.8[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Score.9[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Score.10[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Score.11[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE))

    RepairData_shuffle$AverageRPK[RepairData_shuffle$genes == i] <- mean(sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Reads[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Reads.1[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Reads.2[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Reads.3[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Reads.4[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Reads.5[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Reads.6[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Reads.7[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Reads.8[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Reads.9[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Reads.10[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                   sum(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$Reads.11[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE))/ (max(peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D_shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)/1000)
}

RepairData_shuffle <- RepairData_shuffle[complete.cases(RepairData_shuffle),]
ScaleFactor <- sum(RepairData_shuffle$AverageRPK) / 1000000
RepairData_shuffle$AverageTPM <- RepairData_shuffle$AverageRPK / ScaleFactor

quartiles <- quantile(RepairData_shuffle$AverageTPM)
iqr <- IQR(RepairData_shuffle$AverageTPM)
anomLimit <- as.numeric(quartiles[4]) + iqr*1.5

min <- min(RepairData_shuffle$AverageTPM)
max <- max(RepairData_shuffle$AverageTPM)

ggplot(RepairData_shuffle) + geom_histogram(aes(x=AverageTPM), bins=1000) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + annotate("rect", xmin = min, xmax = quartiles[2], ymin = 0, ymax = 400, alpha = 0.5, fill = "skyblue") + annotate("rect", xmin = anomLimit, xmax = max, ymin = 0, ymax = 400, alpha = 0.5, fill = "darkred") + xlim(0,20) + scale_x_log10()


RepairData_shuffle$Class <- ifelse(RepairData_shuffle$AverageTPM > anomLimit, "Guarded","Normal")
#RepairData_shuffle$Class[RepairData_shuffle$AverageTPM < anomLimit & RepairData_shuffle$AverageTPM > quartiles[4]] <- "Guarded"
RepairData_shuffle$Class[RepairData_shuffle$AverageTPM <= quartiles[2]] <- "Unguarded"


p1 <- ggplot(RepairData_shuffle, aes(length, AverageScore))+
  geom_point(alpha = 0.5, size=.25)+
  xlab("gene length")+
  ylab("Signal")+
  labs() + theme_classic() + scale_x_log10() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + ylim(0,100000)

ggMarginal(p1, type="boxplot")

```

```{r}
p3 <- ggplot(RepairData, aes(length, AverageScore, color=Dataset))+
  geom_point(alpha = 0.25, size=.25)+
  xlab("gene length")+
  ylab("Signal")+
  labs() + theme_classic() + scale_x_log10()  + scale_y_log10() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))

ggMarginal(p3, type="boxplot")
```

```{r}
genes_genic <- as.character(unique(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL))
#genes_shuffle <- as.character(unique(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL))
#genes_shuffle2 <- as.character(unique(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL))


RepairData_Genic <- data.frame(genes_genic)
#RepairData_Shuffle <- data.frame(genes_shuffle)
#RepairData_Shuffle2 <- data.frame(genes_shuffle2)

for(i in RepairData_Genic$genes_genic){
  RepairData_Genic$length[RepairData_Genic$genes_genic == i] <- max(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)
}

for(i in RepairData_Shuffle$genes_shuffle){
  RepairData_Shuffle$length[RepairData_Shuffle$genes_shuffle == i] <- max(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)
}

# for(i in RepairData_Shuffle2$genes_shuffle2){
#   RepairData_Shuffle2$length[RepairData_Shuffle2$genes_shuffle == i] <- max(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)
# }

RepairData_Genic <- RepairData_Genic[is.finite(RepairData_Genic$length),]
RepairData_Shuffle <- RepairData_Shuffle[is.finite(RepairData_Shuffle$length),]
#RepairData_Shuffle2 <- RepairData_Shuffle2[is.finite(RepairData_Shuffle2$length),]

for(i in RepairData_Genic$genes_genic){
  RepairData_Genic$AverageScore[RepairData_Genic$genes_genic == i] <- mean(sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Score[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Score.1[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Score.2[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Score.3[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Score.4[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Score.5[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Score.6[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Score.7[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Score.8[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Score.9[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Score.10[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Score.11[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE))

  RepairData_Genic$AverageRPK[RepairData_Genic$genes_genic == i] <- mean(sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Reads[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Reads.1[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Reads.2[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Reads.3[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Reads.4[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Reads.5[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Reads.6[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Reads.7[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Reads.8[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Reads.9[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Reads.10[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$Reads.11[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE))/ (max(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)/1000)
}


for(i in RepairData_Shuffle$genes_shuffle){
  RepairData_Shuffle$AverageScore[RepairData_Shuffle$genes_shuffle == i] <- mean(sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Score[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Score.1[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Score.2[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Score.3[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Score.4[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Score.5[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Score.6[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Score.7[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Score.8[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Score.9[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Score.10[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                             sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Score.11[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE))
  
  RepairData_Shuffle$AverageRPK[RepairData_Shuffle$genes_shuffle == i] <- mean(sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Reads[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Reads.1[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Reads.2[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Reads.3[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Reads.4[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Reads.5[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Reads.6[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Reads.7[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Reads.8[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Reads.9[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Reads.10[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
                                                                           sum(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$Reads.11[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE))/ (max(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)/1000)
}

# for(i in RepairData_Shuffle2$genes_shuffle2){
#   RepairData_Shuffle2$AverageScore[RepairData_Shuffle2$genes_shuffle == i] <- mean(sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Score[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                              sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Score.1[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                              sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Score.2[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                              sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Score.3[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                              sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Score.4[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                              sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Score.5[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                              sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Score.6[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                              sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Score.7[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                              sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Score.8[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                              sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Score.9[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                              sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Score.10[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                              sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Score.11[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE))
#   
#   RepairData_Shuffle2$AverageRPK[RepairData_Shuffle2$genes_shuffle == i] <- mean(sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Reads[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                            sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Reads.1[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                            sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Reads.2[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                            sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Reads.3[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                            sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Reads.4[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                            sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Reads.5[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                            sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Reads.6[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                            sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Reads.7[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                            sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Reads.8[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                            sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Reads.9[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                            sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Reads.10[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)+
#                                                                            sum(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$Reads.11[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE))/ (max(peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D_Shuffle2@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)/1000)
# }


RepairData_Genic <- RepairData_Genic[complete.cases(RepairData_Genic),]
RepairData_Shuffle <- RepairData_Shuffle[complete.cases(RepairData_Shuffle),]
#RepairData_Shuffle2 <- RepairData_Shuffle2[complete.cases(RepairData_Shuffle2),]



ScaleFactor_genic <- sum(RepairData_Genic$AverageRPK) / 1000000
ScaleFactor_shuffle <- sum(RepairData_Shuffle$AverageRPK) / 1000000
#ScaleFactor_shuffle2 <- sum(RepairData_Shuffle2$AverageRPK) / 1000000


RepairData_Genic$AverageTPM <- RepairData_Genic$AverageRPK / ScaleFactor_genic
RepairData_Shuffle$AverageTPM <- RepairData_Shuffle$AverageRPK / ScaleFactor_shuffle
#RepairData_Shuffle2$AverageTPM <- RepairData_Shuffle2$AverageRPK / ScaleFactor_shuffle2




quartiles_g <- quantile(RepairData_Genic$AverageTPM)
iqr_g <- IQR(RepairData_Genic$AverageTPM)
anomLimit_g <- as.numeric(quartiles_g[4]) + iqr_g*1.5
min_g <- min(RepairData_Genic$AverageTPM)
max_g <- max(RepairData_Genic$AverageTPM)

quartiles_s <- quantile(RepairData_Shuffle$AverageTPM)
iqr_s <- IQR(RepairData_Shuffle$AverageTPM)
anomLimit_s <- as.numeric(quartiles_s[4]) + iqr_s*1.5
min_s <- min(RepairData_Shuffle$AverageTPM)
max_s <- max(RepairData_Shuffle$AverageTPM)

quartiles_s2 <- quantile(RepairData_Shuffle2$AverageTPM)
iqr_s2 <- IQR(RepairData_Shuffle2$AverageTPM)
anomLimit_s2 <- as.numeric(quartiles_s2[4]) + iqr_s2*1.5
min_s2 <- min(RepairData_Shuffle2$AverageTPM)
max_s2 <- max(RepairData_Shuffle2$AverageTPM)




ggplot(RepairData_Genic) + geom_histogram(aes(x=AverageTPM), bins=1000) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + scale_x_log10()
ggplot(RepairData_Shuffle) + geom_histogram(aes(x=AverageTPM), bins=1000) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + scale_x_log10()
ggplot(RepairData_Shuffle2) + geom_histogram(aes(x=AverageTPM), bins=1000) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + scale_x_log10()


#RepairData_Genic$Class <- ifelse(RepairData_Genic$AverageTPM > anomLimit, "Guarded","Normal")
#RepairData_Genic$Class[RepairData_Genic$AverageTPM < anomLimit & RepairData_Genic$AverageTPM > quartiles[4]] <- "Guarded"
#RepairData_Genic$Class[RepairData_Genic$AverageTPM <= quartiles[2]] <- "Unguarded"



#RepairData_Shuffle$Class <- ifelse(RepairData_Shuffle$AverageTPM > anomLimit, "Guarded","Normal")
#RepairData_Shuffle$Class[RepairData_Shuffle$AverageTPM < anomLimit & RepairData_Shuffle$AverageTPM > quartiles[4]] <- "Guarded"
#RepairData_Shuffle$Class[RepairData_Shuffle$AverageTPM <= quartiles[2]] <- "Unguarded"

colnames(RepairData_Genic)[colnames(RepairData_Genic)=="genes_genic"] <- "Gene"
colnames(RepairData_Shuffle)[colnames(RepairData_Shuffle)=="genes_shuffle"] <- "Gene"
colnames(RepairData_Shuffle2)[colnames(RepairData_Shuffle2)=="genes_shuffle2"] <- "Gene"

RepairData_Genic$Dataset <- "Experimental"
RepairData_Shuffle$Dataset <- "Shuffle"
RepairData_Shuffle2$Dataset <- "Shuffle2"


RepairData <- rbind(RepairData_Genic,RepairData_Shuffle)

p3 <- ggplot(new, aes(H1UNG_A, AverageScore))+
geom_point(alpha = 0.2, size=.5)+
xlab("length")+
ylab("Score")+
labs() + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + scale_colour_manual(values= c("darkred","skyblue"))  + scale_x_log10() + scale_y_log10()

ggMarginal(p3, type="density", groupFill=TRUE)
ggplot(Data_New) + geom_point(x=Data_New$AverageScore.x, y=Data_New$AverageScore.y)
```


```{r}
for(i in test_genic$Var1){
  test_genic$length[test_genic$Var1 == i] <- max(peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D_Genic@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)
}

for(i in test_shuffle$Var1){
  test_shuffle$length[test_shuffle$Var1 == i] <- max(peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D_Shuffle@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)
}


test_peakcounts <- rbind(test_genic,test_shuffle)


p3 <- ggplot(test_peakcounts, aes(length, Freq, color=dataset))+
geom_point(alpha = 0.2, size=.5)+
xlab("length")+
ylab("Peak Count")+
labs() + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + scale_colour_manual(values= c("darkred","skyblue"))  + scale_x_log10() + scale_y_log10()
ggMarginal(p3, type="density", groupFill=TRUE)
```

```{r}
peakAnno_1D4D7D_All <- annotatePeak(dba_1D4D7D_Peaks, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")
bed_all <- as.data.frame(peakAnno_1D4D7D_All@anno@elementMetadata@listData)
bedtools_in <- subset(bed_all, select=c(geneChr,geneStart,geneEnd,SYMBOL)) 
data <- unique(bedtools_in)
write.table(data, file="repair_genes.bed", sep="\t", col.names = FALSE, row.names = FALSE)

```


```{r}
setwd("~/RepairSeq/dba_repair_1D4D7D")
load("~/RepairSeq/dba_repair_1D4D7D/dbaset.rda")
load("~/RepairSeq/data/2019-05-08/tpm.rda")

txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
anno86 <- annoGR(EnsDb.Hsapiens.v86)
###USe onyl h1 1d A
dba_1D4D7D_Peaks <- makeGRangesFromDataFrame(as.data.frame(dbaset$peaks[[1]]),keep.extra.columns = TRUE)
peakAnno_1D4D7D <- annotatePeak(dba_1D4D7D_Peaks, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")
dba_1D4D7D_df <- as.data.frame(peakAnno_1D4D7D)
#dba_1D4D7D_df$avgScore <- rowMeans(dba_1D4D7D_df[,c(6,14,22,30,38,46,54,62,70,78,86,94)])
#dba_1D4D7D_df$avgReads <- rowMeans(dba_1D4D7D_df[,c(8,16,24,32,40,48,56,64,72,80,88,96)])


#tpm$ENSEMBL <- row.names(tpm)
#tpm$average <- rowMeans(tpm[,c(1,4,5,10)])

genes_genic <- as.data.frame(table(peakAnno_1D4D7D@anno@elementMetadata@listData$ENSEMBL[peakAnno_1D4D7D@detailGenomicAnnotation$genic == TRUE]))
genes_5UTR <- as.data.frame(table(peakAnno_1D4D7D@anno@elementMetadata@listData$ENSEMBL[peakAnno_1D4D7D@annoStat$Feature == "5' UTR"]))
genes_promoter <- as.data.frame(table(peakAnno_1D4D7D@anno@elementMetadata@listData$ENSEMBL[peakAnno_1D4D7D@annoStat$Feature == "Promoter (<=1kb)"]))



for(i in genes_genic$Var1){
  genes_genic$length[genes_genic$Var1 == i] <- max(peakAnno_1D4D7D@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)
  genes_genic$Score[genes_genic$Var1 == i] <- mean(dba_1D4D7D_df$Score[dba_1D4D7D_df$ENSEMBL == i], na.rm=TRUE)
  genes_genic$RPK[genes_genic$Var1 == i] <- sum(dba_1D4D7D_df$Reads[dba_1D4D7D_df$ENSEMBL == i], na.rm=TRUE) / genes_genic$length[genes_genic$Var1 == i] / 1000
  genes_genic$avgTPM[genes_genic$Var1 == i] <- mean(tpm[row.names(tpm) == i,c(1,4,5,10)])
}

for(i in genes_5UTR$Var1){
  genes_5UTR$length[genes_5UTR$Var1 == i] <- max(peakAnno_1D4D7D@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)
  genes_5UTR$Score[genes_5UTR$Var1 == i] <- sum(dba_1D4D7D_df$Score[dba_1D4D7D_df$ENSEMBL == i], na.rm=TRUE)
  genes_5UTR$RPK[genes_5UTR$Var1 == i] <- sum(dba_1D4D7D_df$Reads[dba_1D4D7D_df$ENSEMBL == i], na.rm=TRUE)  / genes_5UTR$length[genes_5UTR$Var1 == i] / 1000
  genes_5UTR$avgTPM[genes_5UTR$Var1 == i] <- mean(tpm[row.names(tpm) == i,c(1,4,5,10)])

  
} 
for(i in genes_promoter$Var1){
  genes_promoter$length[genes_promoter$Var1 == i] <- max(peakAnno_1D4D7D@anno@elementMetadata@listData$geneLength[peakAnno_1D4D7D@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)
  genes_promoter$Score[genes_promoter$Var1 == i] <- sum(dba_1D4D7D_df$Score[dba_1D4D7D_df$ENSEMBL == i], na.rm=TRUE)
  genes_promoter$RPK[genes_promoter$Var1 == i] <- sum(dba_1D4D7D_df$Reads[dba_1D4D7D_df$ENSEMBL == i], na.rm=TRUE)  / genes_promoter$length[genes_promoter$Var1 == i] / 1000
  genes_promoter$avgTPM[genes_promoter$Var1 == i] <- mean(tpm[row.names(tpm) == i,c(1,4,5,10)])

} 

##RPK sum reads / gene length / 1000
ScaleFactor_genic <- sum(genes_genic$RPK) / 1000000
ScaleFactor_5UTR <- sum(genes_5UTR$RPK) / 1000000
ScaleFactor_promoter <- sum(genes_promoter$RPK) / 1000000

genes_genic$repairTPM <- genes_genic$RPK / ScaleFactor_genic
genes_5UTR$repairTPM <- genes_5UTR$RPK / ScaleFactor_5UTR
genes_promoter$repairTPM <- genes_promoter$RPK / ScaleFactor_promoter


genes_genic$Dataset <- "genic"
genes_5UTR$Dataset <- "5UTR"
genes_promoter$Dataset <- "promoter"


RepairData <- cbind(genes_genic,genes_5UTR,genes_promoter)

```

```{r}
ggplot(genes_genic, aes(avgTPM,repairTPM))+
geom_point(alpha = 0.2, size=.1)+
xlab("RNA Expression (TPM)")+
ylab("Repair (TPM)")+
labs() + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + scale_x_log10() + scale_y_log10()

p3 <- ggplot(RepairData, aes(Score.x, Score.y))+
geom_point(alpha = 0.2, size=.5)+
xlab("Genic_Repair")+
ylab("Promoter_Repair")+
labs() + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) 

ggplot(genes_genic, aes(length, avgReads))+
geom_point(alpha = 0.2, size=.5)+
xlab("length")+
ylab("averageReads")+
labs() + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + scale_x_log10()

```

```{r}
setwd("~/RepairSeq/dba_repair_1D4D7D")
load("~/RepairSeq/dba_repair_1D4D7D/dbaset.rda")
load("~/RepairSeq/data/2019-05-08/tpm.rda")

txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
anno86 <- annoGR(EnsDb.Hsapiens.v86)

H9_1D_A <- dba(dbaset, mask=c(dbaset$masks$ESC_H9))
H9_1D_A <- dba(H9_1D_A, mask=c(H9_1D_A$masks$`1dEdU`))
H9_1D_Peaks <- makeGRangesFromDataFrame(as.data.frame(H9_1D_A$peaks), keep.extra.columns = TRUE)
peakAnno_H9_1D <- annotatePeak(H9_1D_Peaks, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")
peakAnno_H9_1D_df <- as.data.frame(peakAnno_H9_1D)


H91D_ALL <- as.data.frame(table(peakAnno_H9_1D@anno@elementMetadata@listData$ENSEMBL[peakAnno_H9_1D@detailGenomicAnnotation$Intergenic == FALSE]))
H91D_genic <- as.data.frame(table(peakAnno_H9_1D@anno@elementMetadata@listData$ENSEMBL[peakAnno_H9_1D@detailGenomicAnnotation$genic == TRUE]))
H91D_5UTR <- as.data.frame(table(peakAnno_H9_1D@anno@elementMetadata@listData$ENSEMBL[peakAnno_H9_1D@detailGenomicAnnotation$fiveUTR == TRUE]))
H91D_promoter <- as.data.frame(table(peakAnno_H9_1D@anno@elementMetadata@listData$ENSEMBL[peakAnno_H9_1D@detailGenomicAnnotation$Promoter == TRUE]))

for(i in H91D_ALL$Var1){
  H91D_ALL$length[H91D_ALL$Var1 == i] <- max(peakAnno_H9_1D@anno@elementMetadata@listData$geneLength[peakAnno_H9_1D@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)
  H91D_ALL$Score[H91D_ALL$Var1 == i] <- sum(peakAnno_H9_1D_df$Score[peakAnno_H9_1D_df$ENSEMBL == i & peakAnno_H9_1D@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE)
  H91D_ALL$Reads[H91D_ALL$Var1 == i] <- sum(peakAnno_H9_1D_df$Reads[peakAnno_H9_1D_df$ENSEMBL == i & peakAnno_H9_1D@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE)
  H91D_ALL$RPK[H91D_ALL$Var1 == i] <- sum(peakAnno_H9_1D_df$Reads[peakAnno_H9_1D_df$ENSEMBL == i & peakAnno_H9_1D@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE) / H91D_ALL$length[H91D_ALL$Var1 == i] / 1000
}

for(i in H91D_genic$Var1){
  H91D_genic$length[H91D_genic$Var1 == i] <- max(peakAnno_H9_1D@anno@elementMetadata@listData$geneLength[peakAnno_H9_1D@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)
  H91D_genic$Score[H91D_genic$Var1 == i] <- sum(peakAnno_H9_1D_df$Score[peakAnno_H9_1D_df$ENSEMBL == i & peakAnno_H9_1D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)
  H91D_genic$Reads[H91D_genic$Var1 == i] <- sum(peakAnno_H9_1D_df$Reads[peakAnno_H9_1D_df$ENSEMBL == i & peakAnno_H9_1D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE)
  H91D_genic$RPK[H91D_genic$Var1 == i] <- sum(peakAnno_H9_1D_df$Reads[peakAnno_H9_1D_df$ENSEMBL == i & peakAnno_H9_1D@detailGenomicAnnotation$genic == TRUE], na.rm=TRUE) / H91D_genic$length[H91D_genic$Var1 == i] / 1000
}

for(i in H91D_5UTR$Var1){
  H91D_5UTR$length[H91D_5UTR$Var1 == i] <- max(peakAnno_H9_1D@anno@elementMetadata@listData$geneLength[peakAnno_H9_1D@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)
  H91D_5UTR$Score[H91D_5UTR$Var1 == i] <- sum(peakAnno_H9_1D_df$Score[peakAnno_H9_1D_df$ENSEMBL == i & peakAnno_H9_1D@detailGenomicAnnotation$fiveUTR == TRUE], na.rm=TRUE)
  H91D_5UTR$Reads[H91D_5UTR$Var1 == i] <- sum(peakAnno_H9_1D_df$Reads[peakAnno_H9_1D_df$ENSEMBL == i & peakAnno_H9_1D@detailGenomicAnnotation$fiveUTR == TRUE], na.rm=TRUE)
  H91D_5UTR$RPK[H91D_5UTR$Var1 == i] <- sum(peakAnno_H9_1D_df$Reads[peakAnno_H9_1D_df$ENSEMBL == i & peakAnno_H9_1D@detailGenomicAnnotation$fiveUTR == TRUE], na.rm=TRUE) / H91D_5UTR$length[H91D_5UTR$Var1 == i] / 1000
}

for(i in H91D_promoter$Var1){
  H91D_promoter$length[H91D_promoter$Var1 == i] <- max(peakAnno_H9_1D@anno@elementMetadata@listData$geneLength[peakAnno_H9_1D@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)
  H91D_promoter$Score[H91D_promoter$Var1 == i] <- sum(peakAnno_H9_1D_df$Score[peakAnno_H9_1D_df$ENSEMBL == i & peakAnno_H9_1D@detailGenomicAnnotation$Promoter == TRUE], na.rm=TRUE)
  H91D_promoter$Reads[H91D_promoter$Var1 == i] <- sum(peakAnno_H9_1D_df$Reads[peakAnno_H9_1D_df$ENSEMBL == i & peakAnno_H9_1D@detailGenomicAnnotation$Promoter == TRUE], na.rm=TRUE)
  H91D_promoter$RPK[H91D_promoter$Var1 == i] <- sum(peakAnno_H9_1D_df$Reads[peakAnno_H9_1D_df$ENSEMBL == i & peakAnno_H9_1D@detailGenomicAnnotation$Promoter == TRUE], na.rm=TRUE) / 6000 / 1000
}

##RPK sum reads / gene length / 1000
ScaleFactor_All <- sum(H91D_ALL$RPK) / 1000000
ScaleFactor_genic <- sum(H91D_genic$RPK) / 1000000
ScaleFactor_5UTR <- sum(H91D_5UTR$RPK) / 1000000
ScaleFactor_promoter <- sum(H91D_promoter$RPK) / 1000000

H91D_ALL$repairTPM <- H91D_ALL$RPK / ScaleFactor_All
H91D_genic$repairTPM <- H91D_genic$RPK / ScaleFactor_genic
H91D_5UTR$repairTPM <- H91D_5UTR$RPK / ScaleFactor_5UTR
H91D_promoter$repairTPM <- H91D_promoter$RPK / ScaleFactor_promoter

H91D_ALL <- rename(H91D_ALL, c("Var1"="Var1_All", "Freq"="Freq_All","length"="length_All", "Score"="Score_All", "Reads"="Reads_All","RPK"="RPK_All","repairTPM"="repairTPM_All"))
H91D_genic <- rename(H91D_genic, c("Var1"="Var1_genic", "Freq"="Freq_genic","length"="length_genic", "Score"="Score_genic", "Reads"="Reads_genic","RPK"="RPK_genic","repairTPM"="repairTPM_genic"))
H91D_5UTR <- rename(H91D_5UTR, c("Var1"="Var1_5UTR", "Freq"="Freq_5UTR","length"="length_5UTR", "Score"="Score_5UTR", "Reads"="Reads_5UTR","RPK"="RPK_5UTR","repairTPM"="repairTPM_5UTR"))
H91D_promoter <- rename(H91D_promoter, c("Var1"="Var1_promoter", "Freq"="Freq_promoter","length"="length_promoter", "Score"="Score_promoter", "Reads"="Reads_promoter","RPK"="RPK_promoter","repairTPM"="repairTPM_promoter"))


tpm_df <- as.data.frame(tpm)
TPM_H9 <- tpm_df[,c(5,10)]
TPM_H9$ENSEMBL <- rownames(TPM_H9)
TPM_H9$avgTPM <- rowMeans(TPM_H9[c('H9UNG_A', 'H9UNG_B')], na.rm=FALSE)

H9_1D_Data1 <- merge(TPM_H9, H91D_ALL, by.x="ENSEMBL", by.y ="Var1_All", all=TRUE)
H9_1D_Data2 <- merge(H9_1D_Data1, H91D_genic, by.x="ENSEMBL",  by.y ="Var1_genic", all=TRUE)
H9_1D_Data3 <- merge(H9_1D_Data2, H91D_5UTR, by="ENSEMBL", by.y ="Var1_5UTR", all=TRUE)
H9_1D_Data <- merge(H9_1D_Data3, H91D_promoter, by="ENSEMBL",  by.y ="Var1_promoter", all=TRUE)

for(i in H9_1D_Data$ENSEMBL){
  if (length(anno86$gene_name[anno86@ranges@NAMES == i]) > 0) {
   H9_1D_Data$GeneSymbol[H9_1D_Data$ENSEMBL == i] <- anno86$gene_name[anno86@ranges@NAMES == i]
   H9_1D_Data$GeneWidth[H9_1D_Data$ENSEMBL == i] <- anno86@ranges@width[anno86@ranges@NAMES == i]
  }
   else {
     H9_1D_Data$GeneSymbol[H9_1D_Data$ENSEMBL == i] <- i
  }
}
  
  

H9_1D_Data$repairTPM_5UTR[is.na(H9_1D_Data$repairTPM_5UTR)] <- 0
H9_1D_Data$repairTPM_promoter[is.na(H9_1D_Data$repairTPM_promoter)] <- 0
H9_1D_Data$repairTPM_genic[is.na(H9_1D_Data$repairTPM_genic)] <- 0
H9_1D_Data$repairTPM_All[is.na(H9_1D_Data$repairTPM_All)] <- 0

H9_1D_Data$Reads_5UTR[is.na(H9_1D_Data$Reads_5UTR)] <- 0
H9_1D_Data$Reads_promoter[is.na(H9_1D_Data$Reads_promoter)] <- 0
H9_1D_Data$Reads_genic[is.na(H9_1D_Data$Reads_genic)] <- 0
H9_1D_Data$Reads_All[is.na(H9_1D_Data$Reads_All)] <- 0

H9_1D_Data$Score_5UTR[is.na(H9_1D_Data$Score_5UTR)] <- 0
H9_1D_Data$Score_promoter[is.na(H9_1D_Data$Score_promoter)] <- 0
H9_1D_Data$Score_genic[is.na(H9_1D_Data$Score_genic)] <- 0
H9_1D_Data$Score_All[is.na(H9_1D_Data$Score_All)] <- 0

H9_1D_Data$Freq_5UTR[is.na(H9_1D_Data$Freq_5UTR)] <- 0
H9_1D_Data$Freq_promoter[is.na(H9_1D_Data$Freq_promoter)] <- 0
H9_1D_Data$Freq_genic[is.na(H9_1D_Data$Freq_genic)] <- 0
H9_1D_Data$Freq_All[is.na(H9_1D_Data$Freq_All)] <- 0

H9_1D_Data$Class[H9_1D_Data$avgTPM > 0 & H9_1D_Data$repairTPM_genic > 0] <- "Class_4"
H9_1D_Data$Class[H9_1D_Data$avgTPM == 0 & H9_1D_Data$repairTPM_genic > 0] <- "Class_3"
H9_1D_Data$Class[H9_1D_Data$avgTPM > 0 & H9_1D_Data$repairTPM_genic==0] <- "Class_2"
H9_1D_Data$Class[H9_1D_Data$avgTPM == 0 & H9_1D_Data$repairTPM_genic==0] <- "Class_1"

H9_1D_Data$ClassReads[H9_1D_Data$avgTPM > 0 & H9_1D_Data$Reads_genic > 0] <- "Class_4"
H9_1D_Data$ClassReads[H9_1D_Data$avgTPM == 0 & H9_1D_Data$Reads_genic > 0] <- "Class_3"
H9_1D_Data$ClassReads[H9_1D_Data$avgTPM > 0 & H9_1D_Data$Reads_genic==0] <- "Class_2"
H9_1D_Data$ClassReads[H9_1D_Data$avgTPM == 0 & H9_1D_Data$Reads_genic==0] <- "Class_1"

pdf("Repair_Expression_4Class.pdf")
ggplot(H9_1D_Data, aes(x=avgTPM + .0001, y=repairTPM_genic + .0001))  + scale_x_log10() + scale_y_log10() +
geom_point(alpha = 0.2, size=.5, aes(colour=Class))+
xlab("expression TPMs")+
ylab("Repair TPMs")+
theme_classic()  + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))#+ stat_density_2d(aes(fill = ..density..), geom = "raster", contour = FALSE)
dev.off()
#ggMarginal(p3, type="density")

```

```{r}
library(readr)
setwd("~/RepairSeq/dba_repair_1D4D7D")


load("~/RepairSeq/dba_repair_1D4D7D/dbaset.rda")
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
anno86 <- annoGR(EnsDb.Hsapiens.v86)

H9_1D_A <- dba(dbaset, mask=c(dbaset$masks$ESC_H9))
H9_1D_A <- dba(H9_1D_A, mask=c(H9_1D_A$masks$`1dEdU`))
H9_1D_Peaks <- makeGRangesFromDataFrame(as.data.frame(H9_1D_A$peaks), keep.extra.columns = TRUE)
peakAnno_H9_1D <- annotatePeak(H9_1D_Peaks, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")
peakAnno_H9_1D_df <- as.data.frame(peakAnno_H9_1D)

genes_all <- as.character(unique(peakAnno_H9_1D@anno@elementMetadata@listData$SYMBOL))
bed_all <- as.data.frame(peakAnno_H9_1D@anno@elementMetadata@listData)
bedtools_in <- subset(bed_all, select=c(geneChr,geneStart,geneEnd,SYMBOL))
data <- unique(bedtools_in, by=c("SYMBOL"))

write.table(data, file="repair_genes.bed", sep="\t", col.names = FALSE, row.names = FALSE)
write.table(peakAnno_H9_1D_df, file="peakAnno_H9_1D_df.bed", sep="\t", col.names = FALSE, row.names = FALSE)

####bedtools####
library(readr)
setwd("~/RepairSeq/dba_repair_1D4D7D")
H9_1D_A_Genic <- read_delim("H9_1D_Genic_Sort.bed", "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
H9_1D_A_Shuffle <- read_delim("H9_1D_Shuffle_Sort.bed", "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)

colnames(H9_1D_A_Genic) <- colnames(peakAnno_H9_1D_df)
colnames(H9_1D_A_Shuffle) <- colnames(peakAnno_H9_1D_df)

H9_1D_A_Genic_gr <- makeGRangesFromDataFrame(as.data.frame(H9_1D_A_Genic),keep.extra.columns = TRUE)
H9_1D_A_Shuffle_gr <- makeGRangesFromDataFrame(as.data.frame(H9_1D_A_Shuffle),keep.extra.columns = TRUE)

H9_1D_A_Genic_Standard <- keepStandardChromosomes(H9_1D_A_Genic_gr, pruning.mode="coarse")
H9_1D_A_Shuffle_Standard <- keepStandardChromosomes(H9_1D_A_Shuffle_gr, pruning.mode="coarse")

H9_1D_A_Genic_Standard <- dropSeqlevels(H9_1D_A_Genic_Standard, c("chr23","chr24","chrX","chrY"), pruning.mode="coarse")
H9_1D_A_Shuffle_Standard <- dropSeqlevels(H9_1D_A_Shuffle_Standard, c("chr23","chr24","chrX","chrY"), pruning.mode="coarse")


peakAnno_H9_1D_A_Genic <- annotatePeak(H9_1D_A_Genic_Standard, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db",verbose = TRUE)
peakAnno_H9_1D_A_Shuffle <- annotatePeak(H9_1D_A_Shuffle_Standard, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db",verbose = TRUE)

peakAnno_H9_1D_A_Genic_df <- as.data.frame(peakAnno_H9_1D_A_Genic)
peakAnno_H9_1D_A_Shuffle_df <- as.data.frame(peakAnno_H9_1D_A_Shuffle)


Repair_Genic <- as.data.frame(table(peakAnno_H9_1D_A_Genic@anno@elementMetadata@listData$ENSEMBL[peakAnno_H9_1D_A_Genic@detailGenomicAnnotation$Intergenic == FALSE]))
Repair_Shuffle <- as.data.frame(table(peakAnno_H9_1D_A_Shuffle@anno@elementMetadata@listData$ENSEMBL[peakAnno_H9_1D_A_Shuffle@detailGenomicAnnotation$Intergenic == FALSE]))

for(i in Repair_Genic$Var1){
  Repair_Genic$length[Repair_Genic$Var1 == i] <- max(peakAnno_H9_1D_A_Genic@anno@elementMetadata@listData$geneLength[peakAnno_H9_1D_A_Genic@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)
  Repair_Genic$Score[Repair_Genic$Var1 == i] <- sum(peakAnno_H9_1D_A_Genic_df$Score[peakAnno_H9_1D_A_Genic_df$ENSEMBL == i & peakAnno_H9_1D_A_Genic@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE)
  Repair_Genic$Reads[Repair_Genic$Var1 == i] <- sum(peakAnno_H9_1D_A_Genic_df$Reads[peakAnno_H9_1D_A_Genic_df$ENSEMBL == i & peakAnno_H9_1D_A_Genic@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE)
  Repair_Genic$RPK[Repair_Genic$Var1 == i] <- sum(peakAnno_H9_1D_A_Genic_df$Reads[peakAnno_H9_1D_A_Genic_df$ENSEMBL == i & peakAnno_H9_1D_A_Genic@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE) / Repair_Genic$length[Repair_Genic$Var1 == i] / 1000
}

for(i in Repair_Shuffle$Var1){
  Repair_Shuffle$length[Repair_Shuffle$Var1 == i] <- max(peakAnno_H9_1D_A_Shuffle@anno@elementMetadata@listData$geneLength[peakAnno_H9_1D_A_Shuffle@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)
  Repair_Shuffle$Score[Repair_Shuffle$Var1 == i] <- sum(peakAnno_H9_1D_A_Shuffle_df$Score[peakAnno_H9_1D_A_Shuffle_df$ENSEMBL == i & peakAnno_H9_1D_A_Shuffle@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE)
  Repair_Shuffle$Reads[Repair_Shuffle$Var1 == i] <- sum(peakAnno_H9_1D_A_Shuffle_df$Reads[peakAnno_H9_1D_A_Shuffle_df$ENSEMBL == i & peakAnno_H9_1D_A_Shuffle@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE)
  Repair_Shuffle$RPK[Repair_Shuffle$Var1 == i] <- sum(peakAnno_H9_1D_A_Shuffle_df$Reads[peakAnno_H9_1D_A_Shuffle_df$ENSEMBL == i & peakAnno_H9_1D_A_Shuffle@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE) / Repair_Shuffle$length[Repair_Shuffle$Var1 == i] / 1000
}

ScaleFactor_Genic <- sum(Repair_Genic$RPK) / 1000000
ScaleFactor_Shuffle <- sum(Repair_Shuffle$RPK) / 1000000

Repair_Genic$repairTPM <- Repair_Genic$RPK / ScaleFactor_Genic
Repair_Shuffle$repairTPM <- Repair_Shuffle$RPK / ScaleFactor_Shuffle

Repair_Genic <- rename(Repair_Genic, c("Var1"="Var1_Genic", "Freq"="Freq_Genic","length"="length_Genic", "Score"="Score_Genic", "Reads"="Reads_Genic","RPK"="RPK_Genic","repairTPM"="repairTPM_Genic"))
Repair_Shuffle <- rename(Repair_Shuffle, c("Var1"="Var1_Shuffle", "Freq"="Freq_Shuffle","length"="length_Shuffle", "Score"="Score_Shuffle", "Reads"="Reads_Shuffle","RPK"="RPK_Shuffle","repairTPM"="repairTPM_Shuffle"))


RepairData_wShuffle<- merge(Repair_Genic, Repair_Shuffle, by.x="Var1_Genic", by.y ="Var1_Shuffle", all=TRUE)

ggplot(RepairData_wShuffle, aes(repairTPM_Genic, repairTPM_Shuffle))+
xlab("Experimental")+
ylab("Shuffle")+
theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + scale_x_log10() + scale_y_log10() 
```


```{r}
load("~/RepairSeq/dba_samples_AllRepair/dba_samples_AllRepair.rda")
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
anno86 <- annoGR(EnsDb.Hsapiens.v86)
dba_All_1D <- dba(dba_samples_AllRepair, mask=c(dba_samples_AllRepair$masks$`1dEdU`))

dba_damage <- dba.mask(dba_All_1D,DBA_TREATMENT, c("None","ETP","NCS","TOP1","TOP2"), combine="or")

dba_1D_Damage <- dba(dba_All_1D, mask=dba_damage)

dba_Normal <- dba.mask(dba_1D_Damage,DBA_TREATMENT, c("None"))
dba_ETP <- dba.mask(dba_1D_Damage,DBA_TREATMENT, c("ETP"))
dba_NCS <- dba.mask(dba_1D_Damage,DBA_TREATMENT, c("NCS"))
dba_TOP1 <- dba.mask(dba_1D_Damage,DBA_TREATMENT, c("TOP1"))
dba_TOP2 <- dba.mask(dba_1D_Damage,DBA_TREATMENT, c("TOP2"))



#dba_1D_Damage_ETP <- dba.contrast(dba_1D_Damage, dba_Normal, dba_ETP, name1="1D Repair", name2="1D Repair with ETP")  
#dba_1D_Damage_ETP <- dba.analyze(dba_1D_Damage_ETP, method=DBA_DESEQ2)

dba_1D_Damage_NCS <- dba.contrast(dba_1D_Damage, dba_Normal, dba_NCS, name1="1D Repair", name2="1D Repair with NCS")  
dba_1D_Damage_NCS <- dba.analyze(dba_1D_Damage_NCS, method=DBA_DESEQ2)

#dba_1D_Damage_TOP1 <- dba.contrast(dba_1D_Damage, dba_Normal, dba_TOP1, name1="1D Repair", name2="1D Repair with TOP1")  
#dba_1D_Damage_TOP1 <- dba.analyze(dba_1D_Damage_TOP1, method=DBA_DESEQ2)

#dba_1D_Damage_TOP2 <- dba.contrast(dba_1D_Damage, dba_Normal, dba_TOP2, name1="1D Repair", name2="1D Repair with TOP2")  
#dba_1D_Damage_TOP2 <- dba.analyze(dba_1D_Damage_TOP2, method=DBA_DESEQ2)

damageReport_NCS <- dba.report(dba_1D_Damage_NCS, th=1, bCalled = TRUE, bCounts = TRUE, bCalledDetail=TRUE)
peakAnno_NCS <- annotatePeak(damageReport_NCS, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")
peakAnno_NCS_df <- as.data.frame(peakAnno_NCS)
write.table(peakAnno_NCS_df, file="peakAnno_NCS_df.txt", sep="\t", col.names = TRUE, row.names = FALSE)


Repair_NCS <- as.data.frame(table(peakAnno_NCS@anno@elementMetadata@listData$ENSEMBL[peakAnno_NCS@detailGenomicAnnotation$Intergenic == FALSE]))
library("metap")
for(i in Repair_NCS$Var1){
  Repair_NCS$length[Repair_NCS$Var1 == i] <- max(peakAnno_NCS@anno@elementMetadata@listData$geneLength[peakAnno_NCS@anno@elementMetadata@listData$ENSEMBL == i], na.rm=TRUE)
  if (Repair_NCS$Freq[Repair_NCS$Var1 == i] >1) {
  Repair_NCS$avgFold[Repair_NCS$Var1 == i] <- mean(peakAnno_NCS_df$Fold[peakAnno_NCS_df$ENSEMBL == i & peakAnno_NCS@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE)
  Repair_NCS$avgpval[Repair_NCS$Var1 == i] <- sumlog(sort(peakAnno_NCS_df$p.value[peakAnno_NCS_df$ENSEMBL == i & peakAnno_NCS@detailGenomicAnnotation$Intergenic == FALSE]))$p[[1]]
  }
#  else {
#    Repair_NCS$avgFold[Repair_NCS$Var1 == i] <- peakAnno_NCS_df$Fold[peakAnno_NCS_df$ENSEMBL == i & peakAnno_NCS@detailGenomicAnnotation$Intergenic == FALSE]
#    Repair_NCS$sumlog[Repair_NCS$Var1 == i] <- peakAnno_NCS_df$p.value[peakAnno_NCS_df$ENSEMBL == i & peakAnno_NCS@detailGenomicAnnotation$Intergenic == FALSE]
#  }
}

ggplot(Repair_NCS, aes(avgTPM, avgpval))+
geom_contour(alpha = 0.2, size=.5, aes(color=Freq))+
xlab("average Fold Change w/ NCS")+
ylab("combined Pvalue")+
theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5))  +  scale_colour_gradient2(mid = "skyblue", high = "darkred", trans="log10")
```

```{r}
library(readr)
setwd("~/RepairSeq/dba_repair_1D4D7D/")
Walsh_1D4D7D <- read_delim("dba_1D4D7D_All_New_walsh_Anno.bed", "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
Walsh_1D4D7D_Shuffle <- read_delim("dba_1D4D7D_All_Shuffle_New_walsh_Anno.bed", "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)

colnames(Walsh_1D4D7D) <- NULL
colnames(Walsh_1D4D7D)[1:3] <-  c("seqnames","start","end")
#colnames(Walsh_1D4D7D)[1:21] <-  c("seqnames","start","end","ID","Pos","Ref","Alt","X1","X2","X3","X4","X5","x6","Group","Age","Gender","Disease","Region","seqnames.0","start.0","end.0")
#colnames(Walsh_1D4D7D)[22:117] <- colnames(dba_1D4D7D_df)[4:98]

colnames(Walsh_1D4D7D_Shuffle) <- NULL
colnames(Walsh_1D4D7D_Shuffle)[1:3] <-c("seqnames","start","end")
#colnames(Walsh_1D4D7D_Shuffle)[1:21] <-c("seqnames","start","end","ID","Pos","Ref","Alt","X1","X2","X3","X4","X5","x6","Group","Age","Gender","Disease","Region","seqnames.0","start.0","end.0")
#colnames(Walsh_1D4D7D_Shuffle)[22:117] <- colnames(dba_1D4D7D_df)[4:98]

Walsh_1D4D7D$sSNV <- NA
Walsh_1D4D7D_Shuffle$sSNV <- NA

walsh_1D4D7D_gr <- makeGRangesFromDataFrame(as.data.frame(Walsh_1D4D7D),keep.extra.columns = TRUE)
walsh_1D4D7D_Shuffle_gr <- makeGRangesFromDataFrame(Walsh_1D4D7D_Shuffle,keep.extra.columns = TRUE)

colnames(Walsh_1D4D7D)[1:21] <-  c("seqnames","start","end","ID","Pos","Ref","Alt","X1","X2","X3","X4","X5","x6","Group","Age","Gender","Disease","Region","seqnames.0","start.0","end.0")
#colnames(Walsh_1D4D7D)[22:117] <- colnames(dba_1D4D7D_df)[4:98]

colnames(Walsh_1D4D7D_Shuffle)[1:21] <-c("seqnames","start","end","ID","Pos","Ref","Alt","X1","X2","X3","X4","X5","x6","Group","Age","Gender","Disease","Region","seqnames.0","start.0","end.0")
#colnames(Walsh_1D4D7D_Shuffle)[22:117] <- colnames(dba_1D4D7D_df)[4:98]

peakAnno_Walsh_1D4D7D <- annotatePeak(walsh_1D4D7D_gr, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")
peakAnno_Walsh_1D4D7D_df <- as.data.frame(peakAnno_Walsh_1D4D7D)

peakAnno_Walsh_1D4D7D_Shuffle <- annotatePeak(walsh_1D4D7D_Shuffle_gr, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")
peakAnno_Walsh_1D4D7D_Shuffle_df <- as.data.frame(peakAnno_Walsh_1D4D7D_Shuffle)

peakAnno_Walsh_1D4D7D_Shuffle_df$annotation[peakAnno_Walsh_1D4D7D_Shuffle@detailGenomicAnnotation$Exon == "TRUE"] <- "Exon"
peakAnno_Walsh_1D4D7D_Shuffle_df$annotation[peakAnno_Walsh_1D4D7D_Shuffle@detailGenomicAnnotation$Intron == "TRUE"] <- "Intron"
peakAnno_Walsh_1D4D7D_df$annotation[peakAnno_Walsh_1D4D7D@detailGenomicAnnotation$Exon == "TRUE"] <- "Exon"
peakAnno_Walsh_1D4D7D_df$annotation[peakAnno_Walsh_1D4D7D@detailGenomicAnnotation$Intron == "TRUE"] <- "Intron"

peakAnno_Walsh_1D4D7D_df$Data <- "Repair"
peakAnno_Walsh_1D4D7D_Shuffle_df$Data <- "Random_Shuffle"

colnames(peakAnno_Walsh_1D4D7D_Shuffle_df) <- colnames(peakAnno_Walsh_1D4D7D_df)

All_Walsh <- rbind(peakAnno_Walsh_1D4D7D_Shuffle_df, peakAnno_Walsh_1D4D7D_df)
All_Walsh$sSNV[All_Walsh$NA.113 < 1000] <- "sSNV"
All_Walsh$sSNV <- NULL

All_Walsh$NA.108[All_Walsh$NA.108 %like% "Disease"] <- NA


All_Walsh$annotation[All_Walsh$annotation %like% "Downstream"] <- "Downstream"
All_Walsh$annotation[All_Walsh$annotation %like% "Exon"] <- "Exon"
All_Walsh$annotation[All_Walsh$annotation %like% "Intron"] <- "Intron"



All_Walsh_Complete <- na.omit(All_Walsh, cols = c("sSNV"))

#WalshData_forR <- read_csv("WalshData_forR.csv")
#walsh_gr <- makeGRangesFromDataFrame(WalshData_forR,keep.extra.columns = TRUE)

```

```{r}
library(readr)
library(data.table)
setwd("~/RepairSeq/")
Walsh_NCS <- read_delim("peakAnno_NCS_New_Walsh_Anno.bed", "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)

colnames(Walsh_NCS)[1:30] <-  c("seqnames", "start", "end", "width", "strand", "Conc", "Conc_1D.Repair", "Conc_1D.Repair.with.NCS", "Fold", "p.value", "FDR", "H9_1D_A", "H9_1D_B", "H1_1D_A", "H1_1D_B", "H9_UDS_NCS_A", "H9_UDS_NCS_B", "H1_UDS_NCS_A", "H1_UDS_NCS_B", "annotation", "geneChr", "geneStart", "geneEnd", "geneLength", "geneStrand", "geneId", "distanceToTSS", "ENSEMBL", "SYMBOL", "GENENAME")

Walsh_NCS$annotation[Walsh_NCS$annotation %like% "Downstream"] <- "Downstream"
Walsh_NCS$annotation[Walsh_NCS$annotation %like% "Exon"] <- "Exon"
Walsh_NCS$annotation[Walsh_NCS$annotation %like% "Intron"] <- "Intron"
Walsh_NCS$Group <- c()
Walsh_NCS$Group[Walsh_NCS$Fold > 0 & Walsh_NCS$p.value < .05] <- "NCS_Gained"
Walsh_NCS$Group[Walsh_NCS$Fold < 0 & Walsh_NCS$p.value < .05] <- "NCS_Lost"
Walsh_NCS$Group[Walsh_NCS$p.value > .95] <- "Constant"

Walsh_NCS$sSNV <- c()
Walsh_NCS$sSNV[Walsh_NCS$X49 < 1000] <- "sSNV"

Walsh_NCS_Complete <- na.omit(Walsh_NCS, cols = c("sSNV"))



ggplot(Walsh_NCS_Complete, aes(x=X44, fill=Group, y=X49)) + geom_bar(position="dodge", stat = "summary", fun.y = "mean") + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + labs(x = "Neural Class") + labs(y = "Mean Distance to sSNV")

ggplot(Walsh_NCS_Complete, aes(X49,fill=Group)) + geom_histogram(position="dodge", bins=100) + theme_classic() + theme(panel.border = element_rect(colour = "black", fill=NA, size=.5)) + labs(x = "Distance to nearest sSNV") + labs(y = "Number of Peaks") + scale_x_log10() + facet_wrap(~X44)

Repair_NCS <- as.data.frame(table(Walsh_NCS_Complete$SYMBOL))

for(i in Repair_NCS$Var1){
  Repair_NCS$Gain_Count[Repair_NCS$Var1 == i] <- length(Walsh_NCS_Complete$Group[Walsh_NCS_Complete$Group == "NCS_Gained" & Walsh_NCS_Complete$SYMBOL == i])
  Repair_NCS$Loss_Count[Repair_NCS$Var1 == i] <- length(Walsh_NCS_Complete$Group[Walsh_NCS_Complete$Group == "NCS_Lost" & Walsh_NCS_Complete$SYMBOL == i])
  Repair_NCS$Constant_Count[Repair_NCS$Var1 == i] <- length(Walsh_NCS_Complete$Group[Walsh_NCS_Complete$Group == "Constant" & Walsh_NCS_Complete$SYMBOL == i])
}
```


```{r}
library(readr)
setwd("~/RepairSeq/dba_repair_1D4D7D/")
Walsh_1D4D7D_GTEx <- read_delim("dba_1D4D7D_All_New_GTEx_Brain.bed", "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
Walsh_1D4D7D_Shuffle_GTEx <- read_delim("dba_1D4D7D_All_Shuffle_New_GTEx_Brain.bed", "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)

colnames(Walsh_1D4D7D_GTEx) <- NULL
colnames(Walsh_1D4D7D_GTEx)[99:101] <-  c("seqnames","start","end")
#colnames(Walsh_1D4D7D)[1:21] <-  c("seqnames","start","end","ID","Pos","Ref","Alt","X1","X2","X3","X4","X5","x6","Group","Age","Gender","Disease","Region","seqnames.0","start.0","end.0")
#colnames(Walsh_1D4D7D)[22:117] <- colnames(dba_1D4D7D_df)[4:98]

colnames(Walsh_1D4D7D_Shuffle_GTEx) <- NULL
colnames(Walsh_1D4D7D_Shuffle_GTEx)[99:101] <-c("seqnames","start","end")
#colnames(Walsh_1D4D7D_Shuffle)[1:21] <-c("seqnames","start","end","ID","Pos","Ref","Alt","X1","X2","X3","X4","X5","x6","Group","Age","Gender","Disease","Region","seqnames.0","start.0","end.0")
#colnames(Walsh_1D4D7D_Shuffle)[22:117] <- colnames(dba_1D4D7D_df)[4:98]

Walsh_1D4D7D_GTEx$eQTL <- NA
Walsh_1D4D7D_Shuffle_GTEx$eQTL <- NA

walsh_1D4D7D_GTEx_gr <- makeGRangesFromDataFrame(as.data.frame(Walsh_1D4D7D_GTEx),keep.extra.columns = TRUE)
walsh_1D4D7D_Shuffle_GTEx_gr <- makeGRangesFromDataFrame(Walsh_1D4D7D_Shuffle_GTEx,keep.extra.columns = TRUE)

#colnames(Walsh_1D4D7D_GTEx)[1:21] <-  c("seqnames","start","end","ID","Pos","Ref","Alt","X1","X2","X3","X4","X5","x6","Group","Age","Gender","Disease","Region","seqnames.0","start.0","end.0")
#colnames(Walsh_1D4D7D)[22:117] <- colnames(dba_1D4D7D_df)[4:98]

#colnames(Walsh_1D4D7D_Shuffle_GTEx)[1:21] <-c("seqnames","start","end","ID","Pos","Ref","Alt","X1","X2","X3","X4","X5","x6","Group","Age","Gender","Disease","Region","seqnames.0","start.0","end.0")
#colnames(Walsh_1D4D7D_Shuffle)[22:117] <- colnames(dba_1D4D7D_df)[4:98]

peakAnno_Walsh_1D4D7D_GTEx <- annotatePeak(walsh_1D4D7D_GTEx_gr, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")
peakAnno_Walsh_1D4D7D_GTEx_df <- as.data.frame(peakAnno_Walsh_1D4D7D_GTEx)

peakAnno_Walsh_1D4D7D_Shuffle_GTEx <- annotatePeak(walsh_1D4D7D_Shuffle_GTEx_gr, TxDb=txdb, level="gene", annoDb="org.Hs.eg.db")
peakAnno_Walsh_1D4D7D_Shuffle_GTEx_df <- as.data.frame(peakAnno_Walsh_1D4D7D_Shuffle_GTEx)

peakAnno_Walsh_1D4D7D_Shuffle_GTEx_df$annotation[peakAnno_Walsh_1D4D7D_Shuffle_GTEx@detailGenomicAnnotation$Exon == "TRUE"] <- "Exon"
peakAnno_Walsh_1D4D7D_Shuffle_GTEx_df$annotation[peakAnno_Walsh_1D4D7D_Shuffle_GTEx@detailGenomicAnnotation$Intron == "TRUE"] <- "Intron"
peakAnno_Walsh_1D4D7D_GTEx_df$annotation[peakAnno_Walsh_1D4D7D_GTEx@detailGenomicAnnotation$Exon == "TRUE"] <- "Exon"
peakAnno_Walsh_1D4D7D_GTEx_df$annotation[peakAnno_Walsh_1D4D7D_GTEx@detailGenomicAnnotation$Intron == "TRUE"] <- "Intron"

peakAnno_Walsh_1D4D7D_GTEx_df$Data <- "Repair"
peakAnno_Walsh_1D4D7D_Shuffle_GTEx_df$Data <- "Random_Shuffle"

colnames(peakAnno_Walsh_1D4D7D_Shuffle_GTEx_df) <- colnames(peakAnno_Walsh_1D4D7D_GTEx_df)

All_Walsh_GTEx <- rbind(peakAnno_Walsh_1D4D7D_Shuffle_GTEx_df, peakAnno_Walsh_1D4D7D_GTEx_df)
All_Walsh_GTEx$eQTL[All_Walsh_GTEx$NA.112 < 1000] <- "eQTL"
All_Walsh_GTEx$eQTL <- NULL
All_Walsh_GTEx$annotation[All_Walsh_GTEx$annotation %like% "Downstream"] <- "Downstream"
All_Walsh_GTEx$annotation[All_Walsh_GTEx$annotation %like% "Exon"] <- "Exon"
All_Walsh_GTEx$annotation[All_Walsh_GTEx$annotation %like% "Intron"] <- "Intron"


All_Walsh_GTEx_Complete <- na.omit(All_Walsh_GTEx, cols = c("eQTL"))

#WalshData_forR <- read_csv("WalshData_forR.csv")
#walsh_gr <- makeGRangesFromDataFrame(WalshData_forR,keep.extra.columns = TRUE)

```

```{r}
for(i in H3K27Ac_Genic$genes_genic){
  H3K27Ac_Genic$AverageScore[H3K27Ac_Genic$genes_genic == i] <- mean(sum(peakAnno_NoTreatment@anno@elementMetadata@listData$Score.21[peakAnno_NoTreatment@anno@elementMetadata@listData$SYMBOL == i & peakAnno_NoTreatment@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE)+
                                                                       sum(peakAnno_NoTreatment@anno@elementMetadata@listData$Score.24[peakAnno_NoTreatment@anno@elementMetadata@listData$SYMBOL == i & peakAnno_NoTreatment@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE)+
                                                                       sum(peakAnno_NoTreatment@anno@elementMetadata@listData$Score.27[peakAnno_NoTreatment@anno@elementMetadata@listData$SYMBOL == i & peakAnno_NoTreatment@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE)+
                                                                       sum(peakAnno_NoTreatment@anno@elementMetadata@listData$Score.28[peakAnno_NoTreatment@anno@elementMetadata@listData$SYMBOL == i & peakAnno_NoTreatment@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE))
  
  H3K27Ac_Genic$AverageRPK[H3K27Ac_Genic$genes_genic == i] <- mean(sum(peakAnno_NoTreatment@anno@elementMetadata@listData$Reads.21[peakAnno_NoTreatment@anno@elementMetadata@listData$SYMBOL == i & peakAnno_NoTreatment@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE)+
                                                                     sum(peakAnno_NoTreatment@anno@elementMetadata@listData$Reads.24[peakAnno_NoTreatment@anno@elementMetadata@listData$SYMBOL == i & peakAnno_NoTreatment@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE)+
                                                                     sum(peakAnno_NoTreatment@anno@elementMetadata@listData$Reads.27[peakAnno_NoTreatment@anno@elementMetadata@listData$SYMBOL == i & peakAnno_NoTreatment@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE)+
                                                                     sum(peakAnno_NoTreatment@anno@elementMetadata@listData$Reads.28[peakAnno_NoTreatment@anno@elementMetadata@listData$SYMBOL == i & peakAnno_NoTreatment@detailGenomicAnnotation$Intergenic == FALSE], na.rm=TRUE))/max(peakAnno_NoTreatment@anno@elementMetadata@listData$geneLength[peakAnno_NoTreatment@anno@elementMetadata@listData$SYMBOL == i], na.rm=TRUE)/1000 
}

```
